Get data regarding frame count as source, target, use in metaphors, etc.

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("../metanet/data/trimmed/Metaphor.csv")


In [3]:
df

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,label,...,hasExperientialBasis,isRelatedToMetaphorBySource,isAMappingWithinMetaphor,Metaphor_Comments,isADualOfMetaphor,isRelatedToMetaphorByTarget,hasTransitiveSubpart2Metaphor,hasTransitiveSubpart1Metaphor,Metaphor_Experientialbasis,Relatedmetaphor_Description
0,Metaphor_POWER_IS_A_POSSESSED_OBJECT,['Frame_Possession'],['Frame_Power'],"['Example_4540', 'Example_4542', 'Example_4541...",['this metaphor is similar to CONTROL IS A POS...,['Composed/complex'],"['Entailment_196', 'Entailment_195', 'Entailme...",['POWER IS A POSSESSED OBJECT'],['Metaphor_family_Possession_metaphors'],['POWER IS A POSSESSED OBJECT'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,Metaphor_GOVERNMENT_IS_A_SLAVE_MASTER,['Frame_Work_scenario'],['Frame_Government'],"['Example_2625', 'Example_2621', 'Example_2626...",NaN,NaN,NaN,['GOVERNMENT IS A SLAVE MASTER'],['Metaphor_family_Governance_metaphors'],['GOVERNMENT IS A SLAVE MASTER'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,Metaphor_ADDRESSING_POVERTY_IS_TREATING_AN_ILL...,['Frame_Treating_a_physical_affliction'],['Frame_Addressing_poverty'],"['Example_198', 'Example_196', 'Example_199', ...",NaN,"['Entailed', 'Composed/complex']",NaN,['ADDRESSING POVERTY IS TREATING AN ILLNESS'],['Metaphor_family_Physical_afflictions_metapho...,['ADDRESSING POVERTY IS TREATING AN ILLNESS'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,Metaphor_BEING_IMPOVERISHED_IS_BEING_AT_A_LOW_...,['Frame_Being_at_a_Low_Location'],['Frame_Impoverished'],"['Example_587', 'Example_589', 'Example_584', ...",NaN,NaN,"['Entailment_41', 'Entailment_42', 'Entailment...",['BEING IMPOVERISHED IS BEING AT A LOW LOCATION'],"['Metaphor_family_Poverty_metaphors', 'Metapho...",['BEING IMPOVERISHED IS BEING AT A LOW LOCATION'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,Metaphor_TREATIES_ARE_INTERPERSONAL_CONTRACTS,['Frame_Contracts'],['Frame_Treaties'],"['Example_5418', 'Example_5414', 'Example_5412...",NaN,NaN,NaN,['TREATIES ARE INTERPERSONAL CONTRACTS'],['Metaphor_family_Governance_metaphors'],['TREATIES ARE INTERPERSONAL CONTRACTS'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
869,Metaphor_WEALTH_IS_LINEAR_SPATIAL_SCALE,['Frame_Scale'],['Frame_Wealth'],"['Example_5522', 'Example_5523', 'Example_5527...",NaN,NaN,NaN,['WEALTH IS LINEAR SPATIAL SCALE'],['Metaphor_family_Wealth_metaphors'],['WEALTH IS LINEAR SPATIAL SCALE'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
870,Metaphor_FORMING_WORDS_IS_SHAPING,['Frame_Crafting'],['Frame_Forming_words'],"['Example_2195', 'Example_2198', 'Example_2197...",NaN,['Entailed'],NaN,['FORMING WORDS IS SHAPING'],['Metaphor_family_Conduit_Metaphor'],['FORMING WORDS IS SHAPING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
871,Metaphor_COMMUNICATING_IS_POINTING,['Frame_Cause_to_perceive'],['Frame_Communication'],"['Example_1114', 'Example_1113', 'Example_1115...",NaN,['Primary'],NaN,['COMMUNICATING IS POINTING'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",['COMMUNICATING IS POINTING'],...,"[""Pointing to something draws the person's att...",NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
872,Metaphor_INCREASING_EVALUABILITY_OF_GOVERNMENT...,['Frame_TBD'],['Frame_TBD'],"['Example_3172', 'Example_3177', 'Example_3176...",NaN,NaN,NaN,['INCREASING EVALUABILITY OF GOVERNMENT IS INC...,['Metaphor_family_Governance_metaphors'],['INCREASING EVALUABILITY OF GOVERNMENT IS INC...,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [4]:
df.hasSourceFrame.value_counts()

hasSourceFrame
['Frame_TBD']                                    62
['Frame_Verticality']                            28
['Frame_Motion_along_a_path']                    17
['Frame_Machine']                                14
['Frame_Possession']                             13
                                                 ..
['Frame_Clear_paths_to_accessing_a_location']     1
['Frame_Actor']                                   1
['Frame_Equalizing_quantities']                   1
['Frame_Caused_downward_motion']                  1
['Frame_Cause_to_perceive']                       1
Name: count, Length: 305, dtype: int64

In [5]:
df.hasTargetFrame.value_counts()

hasTargetFrame
['Frame_TBD']                          62
['Frame_Nation']                       22
['Frame_Government_institution']       17
['Frame_Taxation']                     16
['Frame_Improving_economic_status']    16
                                       ..
['Frame_Plurality']                     1
['Frame_Spatial_relation']              1
['Frame_Positivity_scale']              1
['Frame_Limited_resource']              1
['Frame_Forming_words']                 1
Name: count, Length: 261, dtype: int64

In [6]:
counts = pd.concat(
    [df["hasSourceFrame"], df["hasTargetFrame"]],
    ignore_index=True
).dropna().value_counts()

In [7]:
counts

['Frame_TBD']                        124
['Frame_Verticality']                 28
['Frame_Nation']                      24
['Frame_Government_institution']      17
['Frame_Motion_along_a_path']         17
                                    ... 
['Frame_Body_weight']                  1
['Frame_Paths_out_of_a_location']      1
['Frame_Concentration']                1
['Frame_Sentence']                     1
['Frame_Forming_words']                1
Name: count, Length: 526, dtype: int64

In [8]:
df.hasMetaphorType.value_counts()

hasMetaphorType
['Composed/complex']                           278
['Primary']                                    137
['Primary', 'Entailed']                         23
['Entailed']                                    20
['Composed/complex', 'Entailed']                17
['Entailed', 'Composed/complex']                16
['Entailed', 'Primary']                         10
['Primary', 'Composed/complex']                  5
['Composed/complex', 'Primary']                  2
['Composed/complex', 'Entailed', 'Primary']      1
['Primary', 'Entailed', 'Composed/complex']      1
Name: count, dtype: int64

There are 510 total metaphors in MetaNet. Of these, 179 of the metaphors include a label of "Primary" among the listed metaphor types. 137 metaphors are labeled exclusively as Primary. These will be assessed to confirm that the metaphor types do not cause conflict.

In [9]:
primary = df[df["hasMetaphorType"].str.contains("Primary", na=False)]

In [10]:
primary

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,label,...,hasExperientialBasis,isRelatedToMetaphorBySource,isAMappingWithinMetaphor,Metaphor_Comments,isADualOfMetaphor,isRelatedToMetaphorByTarget,hasTransitiveSubpart2Metaphor,hasTransitiveSubpart1Metaphor,Metaphor_Experientialbasis,Relatedmetaphor_Description
6,Metaphor_ACHIEVING_A_PURPOSE_IS_REACHING_A_DES...,['Frame_Motion_to_a_location'],['Frame_Purposeful_action'],"['Example_91', 'Example_92', 'Example_93', 'Ex...",NaN,"['Primary', 'Entailed']","['Entailment_8', 'Entailment_7', 'Entailment_9']",['ACHIEVING A PURPOSE IS REACHING A DESTINATION'],"['Metaphor_family_Grady_Primary_Metaphors', 'M...",['ACHIEVING A PURPOSE IS REACHING A DESTINATION'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
10,Metaphor_PROGRESS_IS_FORWARD_MOTION_ALONG_A_PATH,['Frame_Motion_along_a_path'],['Frame_Action'],"['Example_4628', 'Example_4627', 'Example_4630...",NaN,"['Primary', 'Entailed']",NaN,['PROGRESS IS FORWARD MOTION ALONG A PATH'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",['PROGRESS IS FORWARD MOTION ALONG A PATH'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
12,Metaphor_DISTINGUISHING_IS_DIVIDING,['Frame_Dividing'],['Frame_Distinguishing'],"['Example_1700', 'Example_1701', 'Example_1705...",NaN,['Primary'],NaN,['DISTINGUISHING IS DIVIDING'],['Metaphor_family_Primary_Metaphors'],['DISTINGUISHING IS DIVIDING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
18,Metaphor_ACTION_IS_MOTION,['Frame_Motion'],['Frame_Action'],"['Example_136', 'Example_134', 'Example_132', ...","[""What is the appropriate Source frame here? ...",['Primary'],NaN,['ACTION IS MOTION'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",['ACTION IS MOTION'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
26,Metaphor_ACQUIRING_IDEAS_IS_EATING,['Frame_Eating'],['Frame_Thinking_about'],"['Example_108', 'Example_115', 'Example_107', ...",['See the numerous subcases of this metaphor f...,['Primary'],['Entailment_10'],['ACQUIRING IDEAS IS EATING'],"['Metaphor_family_Mind_Metaphors', 'Metaphor_f...",['ACQUIRING IDEAS IS EATING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
845,Metaphor_ATTRIBUTES_ARE_POSSESSIONS,['Frame_Possession'],['Frame_Attribute'],"['Example_523', 'Example_524', 'Example_521', ...",NaN,['Primary'],"['Entailment_38', 'Entailment_37', 'Entailment...",['ATTRIBUTES ARE POSSESSIONS'],['Metaphor_family_Object_Event_Structure_Metap...,['ATTRIBUTES ARE POSSESSIONS'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
850,Metaphor_MEANS_ARE_PATHS,['Frame_Self_propelled_motion_to_a_destination'],['Frame_Purposeful_action'],"['Example_3701', 'Example_3702', 'Example_3707...",NaN,"['Entailed', 'Primary']",['Entailment_165'],['MEANS ARE PATHS'],"['Metaphor_family_Grady_Primary_Metaphors', 'M...",['MEANS ARE PATHS'],...,['The correlation between goal-oriented decisi...,NaN,['Metaphor_ACHIEVING_A_PURPOSE_IS_REACHING_A_D...,NaN,NaN,NaN,NaN,NaN,NaN,NaN
861,Metaphor_EXPERIENCE_OF_A_STATE_IS_SPATIAL_RELA...,['Frame_Spatial_relation'],['Frame_Experiencing_a_state'],"['Example_2101', 'Example_2098', 'Example_2097...",['This is a non-perspectivized view of the spa...,['Primary'],"['Entailment_93', 'Entailment_92', 'Entailment...",['EXPERIENCE OF A STATE IS SPATIAL RELATION WI...,"['Metaphor_family_Orientational_Metaphors', 'M...",['EXPERIENCE OF A STATE IS SPATIAL RELATION WI...,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
864,Metaphor_BECOMING_AWARE_IS_NOTICING,['Frame_Noticing'],['Frame_Becoming_aware'],"['Example_555', 'Example_556', 'Example_553', ...",NaN,['Primary'],NaN,['BECOMING AWARE IS NOTICING'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",['BECOMING AWARE IS NOTICING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [11]:
primary.hasSourceFrame.value_counts()

hasSourceFrame
['Frame_Verticality']            20
['Frame_Object']                  7
['Frame_Motion_along_a_path']     7
['Frame_Bounded_region']          6
['Frame_Being_at_a_location']     5
                                 ..
['Frame_Aids_to_motion']          1
['Frame_Front']                   1
['Frame_Weight']                  1
['Frame_Digestion']               1
['Frame_Cause_to_perceive']       1
Name: count, Length: 95, dtype: int64

In [12]:
primary.hasTargetFrame.value_counts()

hasTargetFrame
['Frame_Time']                 8
['Frame_Action']               8
['Frame_Emotion']              7
['Frame_Quantity']             7
['Frame_Purposeful_action']    5
                              ..
['Frame_Life']                 1
['Frame_Change_of_state']      1
['Frame_Social_group']         1
['Frame_Assistance']           1
['Frame_Becoming_aware']       1
Name: count, Length: 96, dtype: int64

In [13]:
import ast

def is_primary_only(x):
    try:
        return ast.literal_eval(x) == ["Primary"]
    except Exception:
        return False

primary_only = df[df["hasMetaphorType"].apply(is_primary_only)]


In [14]:
primary_only

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,label,...,hasExperientialBasis,isRelatedToMetaphorBySource,isAMappingWithinMetaphor,Metaphor_Comments,isADualOfMetaphor,isRelatedToMetaphorByTarget,hasTransitiveSubpart2Metaphor,hasTransitiveSubpart1Metaphor,Metaphor_Experientialbasis,Relatedmetaphor_Description
12,Metaphor_DISTINGUISHING_IS_DIVIDING,['Frame_Dividing'],['Frame_Distinguishing'],"['Example_1700', 'Example_1701', 'Example_1705...",NaN,['Primary'],NaN,['DISTINGUISHING IS DIVIDING'],['Metaphor_family_Primary_Metaphors'],['DISTINGUISHING IS DIVIDING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
18,Metaphor_ACTION_IS_MOTION,['Frame_Motion'],['Frame_Action'],"['Example_136', 'Example_134', 'Example_132', ...","[""What is the appropriate Source frame here? ...",['Primary'],NaN,['ACTION IS MOTION'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",['ACTION IS MOTION'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
26,Metaphor_ACQUIRING_IDEAS_IS_EATING,['Frame_Eating'],['Frame_Thinking_about'],"['Example_108', 'Example_115', 'Example_107', ...",['See the numerous subcases of this metaphor f...,['Primary'],['Entailment_10'],['ACQUIRING IDEAS IS EATING'],"['Metaphor_family_Mind_Metaphors', 'Metaphor_f...",['ACQUIRING IDEAS IS EATING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
33,Metaphor_BEING_WEALTHY_IS_BEING_FREE,['Frame_Lack_of_impediment_to_motion'],['Frame_Wealth'],"['Example_647', 'Example_649', 'Example_650', ...",NaN,['Primary'],NaN,['BEING WEALTHY IS BEING FREE'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",['BEING WEALTHY IS BEING FREE'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
38,Metaphor_COMMUNICATING_IS_SHOWING,['Frame_Cause_to_see'],['Frame_Communication'],"['Example_1124', 'Example_1117', 'Example_1121...",NaN,['Primary'],['Entailment_66'],['COMMUNICATING IS SHOWING'],"['Metaphor_family_Conduit_Metaphor', 'Metaphor...",['COMMUNICATING IS SHOWING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
844,Metaphor_HAPPINESS_IS_VERTICALITY,['Frame_Verticality'],['Frame_Happiness'],"['Example_2821', 'Example_2822', 'Example_2818...",NaN,['Primary'],"['Entailment_123', 'Entailment_124']",['HAPPINESS IS VERTICALITY'],"['Metaphor_family_Happiness_Metaphors', 'Metap...",['HAPPINESS IS VERTICALITY'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
845,Metaphor_ATTRIBUTES_ARE_POSSESSIONS,['Frame_Possession'],['Frame_Attribute'],"['Example_523', 'Example_524', 'Example_521', ...",NaN,['Primary'],"['Entailment_38', 'Entailment_37', 'Entailment...",['ATTRIBUTES ARE POSSESSIONS'],['Metaphor_family_Object_Event_Structure_Metap...,['ATTRIBUTES ARE POSSESSIONS'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
861,Metaphor_EXPERIENCE_OF_A_STATE_IS_SPATIAL_RELA...,['Frame_Spatial_relation'],['Frame_Experiencing_a_state'],"['Example_2101', 'Example_2098', 'Example_2097...",['This is a non-perspectivized view of the spa...,['Primary'],"['Entailment_93', 'Entailment_92', 'Entailment...",['EXPERIENCE OF A STATE IS SPATIAL RELATION WI...,"['Metaphor_family_Orientational_Metaphors', 'M...",['EXPERIENCE OF A STATE IS SPATIAL RELATION WI...,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
864,Metaphor_BECOMING_AWARE_IS_NOTICING,['Frame_Noticing'],['Frame_Becoming_aware'],"['Example_555', 'Example_556', 'Example_553', ...",NaN,['Primary'],NaN,['BECOMING AWARE IS NOTICING'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",['BECOMING AWARE IS NOTICING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [15]:
primary_only.hasSourceFrame.value_counts()


hasSourceFrame
['Frame_Verticality']            10
['Frame_Object']                  7
['Frame_Bounded_region']          6
['Frame_Being_at_a_location']     5
['Frame_Motion_along_a_path']     4
                                 ..
['Frame_Making']                  1
['Frame_Smelly']                  1
['Frame_Fluid_motion']            1
['Frame_Measuring']               1
['Frame_Cause_to_perceive']       1
Name: count, Length: 82, dtype: int64

In [16]:
primary_only.hasTargetFrame.value_counts()


hasTargetFrame
['Frame_Time']                      6
['Frame_Action']                    4
['Frame_Causation']                 4
['Frame_Difficulties_in_action']    4
['Frame_Thinking_about']            3
                                   ..
['Frame_Change_of_state']           1
['Frame_Sympathy']                  1
['Frame_Organization']              1
['Frame_Abstract_relatedness']      1
['Frame_Becoming_aware']            1
Name: count, Length: 89, dtype: int64

In [17]:
primary_only

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,label,...,hasExperientialBasis,isRelatedToMetaphorBySource,isAMappingWithinMetaphor,Metaphor_Comments,isADualOfMetaphor,isRelatedToMetaphorByTarget,hasTransitiveSubpart2Metaphor,hasTransitiveSubpart1Metaphor,Metaphor_Experientialbasis,Relatedmetaphor_Description
12,Metaphor_DISTINGUISHING_IS_DIVIDING,['Frame_Dividing'],['Frame_Distinguishing'],"['Example_1700', 'Example_1701', 'Example_1705...",NaN,['Primary'],NaN,['DISTINGUISHING IS DIVIDING'],['Metaphor_family_Primary_Metaphors'],['DISTINGUISHING IS DIVIDING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
18,Metaphor_ACTION_IS_MOTION,['Frame_Motion'],['Frame_Action'],"['Example_136', 'Example_134', 'Example_132', ...","[""What is the appropriate Source frame here? ...",['Primary'],NaN,['ACTION IS MOTION'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",['ACTION IS MOTION'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
26,Metaphor_ACQUIRING_IDEAS_IS_EATING,['Frame_Eating'],['Frame_Thinking_about'],"['Example_108', 'Example_115', 'Example_107', ...",['See the numerous subcases of this metaphor f...,['Primary'],['Entailment_10'],['ACQUIRING IDEAS IS EATING'],"['Metaphor_family_Mind_Metaphors', 'Metaphor_f...",['ACQUIRING IDEAS IS EATING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
33,Metaphor_BEING_WEALTHY_IS_BEING_FREE,['Frame_Lack_of_impediment_to_motion'],['Frame_Wealth'],"['Example_647', 'Example_649', 'Example_650', ...",NaN,['Primary'],NaN,['BEING WEALTHY IS BEING FREE'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",['BEING WEALTHY IS BEING FREE'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
38,Metaphor_COMMUNICATING_IS_SHOWING,['Frame_Cause_to_see'],['Frame_Communication'],"['Example_1124', 'Example_1117', 'Example_1121...",NaN,['Primary'],['Entailment_66'],['COMMUNICATING IS SHOWING'],"['Metaphor_family_Conduit_Metaphor', 'Metaphor...",['COMMUNICATING IS SHOWING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
844,Metaphor_HAPPINESS_IS_VERTICALITY,['Frame_Verticality'],['Frame_Happiness'],"['Example_2821', 'Example_2822', 'Example_2818...",NaN,['Primary'],"['Entailment_123', 'Entailment_124']",['HAPPINESS IS VERTICALITY'],"['Metaphor_family_Happiness_Metaphors', 'Metap...",['HAPPINESS IS VERTICALITY'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
845,Metaphor_ATTRIBUTES_ARE_POSSESSIONS,['Frame_Possession'],['Frame_Attribute'],"['Example_523', 'Example_524', 'Example_521', ...",NaN,['Primary'],"['Entailment_38', 'Entailment_37', 'Entailment...",['ATTRIBUTES ARE POSSESSIONS'],['Metaphor_family_Object_Event_Structure_Metap...,['ATTRIBUTES ARE POSSESSIONS'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
861,Metaphor_EXPERIENCE_OF_A_STATE_IS_SPATIAL_RELA...,['Frame_Spatial_relation'],['Frame_Experiencing_a_state'],"['Example_2101', 'Example_2098', 'Example_2097...",['This is a non-perspectivized view of the spa...,['Primary'],"['Entailment_93', 'Entailment_92', 'Entailment...",['EXPERIENCE OF A STATE IS SPATIAL RELATION WI...,"['Metaphor_family_Orientational_Metaphors', 'M...",['EXPERIENCE OF A STATE IS SPATIAL RELATION WI...,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
864,Metaphor_BECOMING_AWARE_IS_NOTICING,['Frame_Noticing'],['Frame_Becoming_aware'],"['Example_555', 'Example_556', 'Example_553', ...",NaN,['Primary'],NaN,['BECOMING AWARE IS NOTICING'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",['BECOMING AWARE IS NOTICING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [18]:
# === 1) produce primary_frames.txt ===
import ast
import re
import pandas as pd

# If df isn't already loaded, load it:
# df = pd.read_csv("path/to/your.csv")

def is_primary_only(x):
    """Return True only if x represents exactly ['Primary'] or the string 'Primary'."""
    if pd.isna(x):
        return False
    if x == "Primary":
        return True
    if isinstance(x, list):
        return x == ["Primary"]
    if isinstance(x, str):
        # Try literal_eval for list-like strings: "['Primary']"
        try:
            val = ast.literal_eval(x)
            return val == ["Primary"]
        except Exception:
            # Fall back to delimiter-splitting (e.g. "Primary; Complex" -> not exclusive)
            parts = re.split(r'[;,]\s*', x.strip())
            parts = [p for p in parts if p]
            return len(parts) == 1 and parts[0] == "Primary"
    return False

# mask rows that are exclusively Primary
mask = df["hasMetaphorType"].apply(is_primary_only)

# collect frames that appear in source or target columns within those rows
src = df.loc[mask, "hasSourceFrame"].dropna()
tgt = df.loc[mask, "hasTargetFrame"].dropna()


def normalize_frame_name(x):
    # Case 1: actual list
    if isinstance(x, list):
        if len(x) != 1:
            return None
        x = x[0]

    # Case 2: list-like string: "['frame_action']"
    if isinstance(x, str):
        try:
            val = ast.literal_eval(x)
            if isinstance(val, list) and len(val) == 1:
                x = val[0]
        except Exception:
            pass

    if not isinstance(x, str):
        return None

    x = x.strip().lower()
    if x.startswith("frame_"):
        x = x[len("frame_"):]
    return x


primary_frames = (
    pd.concat([src, tgt])
      .map(normalize_frame_name)
      .dropna()
      .unique()
)

primary_set = set(primary_frames)

with open("data/primary_frames.txt", "w", encoding="utf-8") as fw:
    for f in sorted(primary_set):
        fw.write(f + "\n")

print(f"Wrote {len(primary_set)} frames to primary_frames.txt")


Wrote 165 frames to primary_frames.txt


In [19]:
# === 2) filter the wordlists file by the primary frames ===
import re

INPUT = "data/word_lists.txt"                   # original file (Frame_...: format)
OUTPUT = "data/word_lists_primary_only.txt"     # filtered output

frame_header_re = re.compile(r'^Frame_(.+?):\s*$')
pos_re = re.compile(r'\.[A-Za-z]+$')            # .n .v .a etc.
lead_marker_re = re.compile(r'^[\-\*\•\s]+')    # drop bullets/spaces

with open(INPUT, "r", encoding="utf-8") as fin, \
     open(OUTPUT, "w", encoding="utf-8") as fout:

    current_frame = None
    include_current = False
    words = []
    seen = set()

    for raw in fin:
        line = raw.strip()
        if not line:
            continue

        m = frame_header_re.match(line)
        if m:
            # flush previous frame if included
            if include_current and words:
                fout.write(current_frame + "\n")
                for w in words:
                    fout.write(w + "\n")
                fout.write("\n")

            current_frame = m.group(1).lower()
            include_current = current_frame in primary_set
            words = []
            seen = set()
            continue

        if not include_current:
            continue

        token = lead_marker_re.sub('', line)
        # skip multiword entries
        if " " in token:
            continue
        token = pos_re.sub('', token).strip()
        if not token:
            continue
        if token in seen:
            continue
        seen.add(token)
        words.append(token)

    # final flush
    if include_current and words:
        fout.write(current_frame + "\n")
        for w in words:
            fout.write(w + "\n")
        fout.write("\n")

print("Wrote filtered wordlists to:", OUTPUT)


Wrote filtered wordlists to: data/word_lists_primary_only.txt


In [20]:
primary_only

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,label,...,hasExperientialBasis,isRelatedToMetaphorBySource,isAMappingWithinMetaphor,Metaphor_Comments,isADualOfMetaphor,isRelatedToMetaphorByTarget,hasTransitiveSubpart2Metaphor,hasTransitiveSubpart1Metaphor,Metaphor_Experientialbasis,Relatedmetaphor_Description
12,Metaphor_DISTINGUISHING_IS_DIVIDING,['Frame_Dividing'],['Frame_Distinguishing'],"['Example_1700', 'Example_1701', 'Example_1705...",NaN,['Primary'],NaN,['DISTINGUISHING IS DIVIDING'],['Metaphor_family_Primary_Metaphors'],['DISTINGUISHING IS DIVIDING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
18,Metaphor_ACTION_IS_MOTION,['Frame_Motion'],['Frame_Action'],"['Example_136', 'Example_134', 'Example_132', ...","[""What is the appropriate Source frame here? ...",['Primary'],NaN,['ACTION IS MOTION'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",['ACTION IS MOTION'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
26,Metaphor_ACQUIRING_IDEAS_IS_EATING,['Frame_Eating'],['Frame_Thinking_about'],"['Example_108', 'Example_115', 'Example_107', ...",['See the numerous subcases of this metaphor f...,['Primary'],['Entailment_10'],['ACQUIRING IDEAS IS EATING'],"['Metaphor_family_Mind_Metaphors', 'Metaphor_f...",['ACQUIRING IDEAS IS EATING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
33,Metaphor_BEING_WEALTHY_IS_BEING_FREE,['Frame_Lack_of_impediment_to_motion'],['Frame_Wealth'],"['Example_647', 'Example_649', 'Example_650', ...",NaN,['Primary'],NaN,['BEING WEALTHY IS BEING FREE'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",['BEING WEALTHY IS BEING FREE'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
38,Metaphor_COMMUNICATING_IS_SHOWING,['Frame_Cause_to_see'],['Frame_Communication'],"['Example_1124', 'Example_1117', 'Example_1121...",NaN,['Primary'],['Entailment_66'],['COMMUNICATING IS SHOWING'],"['Metaphor_family_Conduit_Metaphor', 'Metaphor...",['COMMUNICATING IS SHOWING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
844,Metaphor_HAPPINESS_IS_VERTICALITY,['Frame_Verticality'],['Frame_Happiness'],"['Example_2821', 'Example_2822', 'Example_2818...",NaN,['Primary'],"['Entailment_123', 'Entailment_124']",['HAPPINESS IS VERTICALITY'],"['Metaphor_family_Happiness_Metaphors', 'Metap...",['HAPPINESS IS VERTICALITY'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
845,Metaphor_ATTRIBUTES_ARE_POSSESSIONS,['Frame_Possession'],['Frame_Attribute'],"['Example_523', 'Example_524', 'Example_521', ...",NaN,['Primary'],"['Entailment_38', 'Entailment_37', 'Entailment...",['ATTRIBUTES ARE POSSESSIONS'],['Metaphor_family_Object_Event_Structure_Metap...,['ATTRIBUTES ARE POSSESSIONS'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
861,Metaphor_EXPERIENCE_OF_A_STATE_IS_SPATIAL_RELA...,['Frame_Spatial_relation'],['Frame_Experiencing_a_state'],"['Example_2101', 'Example_2098', 'Example_2097...",['This is a non-perspectivized view of the spa...,['Primary'],"['Entailment_93', 'Entailment_92', 'Entailment...",['EXPERIENCE OF A STATE IS SPATIAL RELATION WI...,"['Metaphor_family_Orientational_Metaphors', 'M...",['EXPERIENCE OF A STATE IS SPATIAL RELATION WI...,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
864,Metaphor_BECOMING_AWARE_IS_NOTICING,['Frame_Noticing'],['Frame_Becoming_aware'],"['Example_555', 'Example_556', 'Example_553', ...",NaN,['Primary'],NaN,['BECOMING AWARE IS NOTICING'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",['BECOMING AWARE IS NOTICING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [21]:
primary_cleaned = primary_only[['name', 'hasSourceFrame', 'hasTargetFrame', 'hasExample', 'comment', 'hasMetaphorType', 'hasEntailment', 'hasName', 'isInMetaphorFamily', 'hasMappings', 'hasAlias']]
primary_cleaned

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,hasMappings,hasAlias
12,Metaphor_DISTINGUISHING_IS_DIVIDING,['Frame_Dividing'],['Frame_Distinguishing'],"['Example_1700', 'Example_1701', 'Example_1705...",NaN,['Primary'],NaN,['DISTINGUISHING IS DIVIDING'],['Metaphor_family_Primary_Metaphors'],NaN,['DISTINGUISHING IS DIVIDING(Sweetser (1987):n...
18,Metaphor_ACTION_IS_MOTION,['Frame_Motion'],['Frame_Action'],"['Example_136', 'Example_134', 'Example_132', ...","[""What is the appropriate Source frame here? ...",['Primary'],NaN,['ACTION IS MOTION'],"['Metaphor_family_Primary_Metaphors', 'Metapho...","['Mapping_25', 'Mapping_24']","['ACTION IS MOTION(null:null)', 'ACTION IS MOV..."
26,Metaphor_ACQUIRING_IDEAS_IS_EATING,['Frame_Eating'],['Frame_Thinking_about'],"['Example_108', 'Example_115', 'Example_107', ...",['See the numerous subcases of this metaphor f...,['Primary'],['Entailment_10'],['ACQUIRING IDEAS IS EATING'],"['Metaphor_family_Mind_Metaphors', 'Metaphor_f...","['Mapping_19', 'Mapping_20', 'Mapping_21']",['ACQUIRING IDEAS IS EATING(null:null)']
33,Metaphor_BEING_WEALTHY_IS_BEING_FREE,['Frame_Lack_of_impediment_to_motion'],['Frame_Wealth'],"['Example_647', 'Example_649', 'Example_650', ...",NaN,['Primary'],NaN,['BEING WEALTHY IS BEING FREE'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",NaN,['BEING WEALTHY IS BEING FREE(null:null)']
38,Metaphor_COMMUNICATING_IS_SHOWING,['Frame_Cause_to_see'],['Frame_Communication'],"['Example_1124', 'Example_1117', 'Example_1121...",NaN,['Primary'],['Entailment_66'],['COMMUNICATING IS SHOWING'],"['Metaphor_family_Conduit_Metaphor', 'Metaphor...","['Mapping_176', 'Mapping_174', 'Mapping_175', ...",['COMMUNICATING IS SHOWING(null:null)']
...,...,...,...,...,...,...,...,...,...,...,...
844,Metaphor_HAPPINESS_IS_VERTICALITY,['Frame_Verticality'],['Frame_Happiness'],"['Example_2821', 'Example_2822', 'Example_2818...",NaN,['Primary'],"['Entailment_123', 'Entailment_124']",['HAPPINESS IS VERTICALITY'],"['Metaphor_family_Happiness_Metaphors', 'Metap...",NaN,"['HAPPINESS IS VERTICALITY(Metaphor+:null)', '..."
845,Metaphor_ATTRIBUTES_ARE_POSSESSIONS,['Frame_Possession'],['Frame_Attribute'],"['Example_523', 'Example_524', 'Example_521', ...",NaN,['Primary'],"['Entailment_38', 'Entailment_37', 'Entailment...",['ATTRIBUTES ARE POSSESSIONS'],['Metaphor_family_Object_Event_Structure_Metap...,"['Mapping_113', 'Mapping_114']",['ATTRIBUTES ARE POSSESSIONS(Grady (1998):null...
861,Metaphor_EXPERIENCE_OF_A_STATE_IS_SPATIAL_RELA...,['Frame_Spatial_relation'],['Frame_Experiencing_a_state'],"['Example_2101', 'Example_2098', 'Example_2097...",['This is a non-perspectivized view of the spa...,['Primary'],"['Entailment_93', 'Entailment_92', 'Entailment...",['EXPERIENCE OF A STATE IS SPATIAL RELATION WI...,"['Metaphor_family_Orientational_Metaphors', 'M...","['Mapping_256', 'Mapping_257', 'Mapping_258']",['EXPERIENCE OF A STATE IS SPATIAL RELATION WI...
864,Metaphor_BECOMING_AWARE_IS_NOTICING,['Frame_Noticing'],['Frame_Becoming_aware'],"['Example_555', 'Example_556', 'Example_553', ...",NaN,['Primary'],NaN,['BECOMING AWARE IS NOTICING'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",NaN,NaN


In [22]:
import ast

primary_cleaned['hasSourceFrame'] = primary_cleaned['hasSourceFrame'].apply(lambda x: ast.literal_eval(x)[0])
primary_cleaned['hasTargetFrame'] = primary_cleaned['hasTargetFrame'].apply(lambda x: ast.literal_eval(x)[0])
primary_cleaned['hasMetaphorType'] = primary_cleaned['hasMetaphorType'].apply(lambda x: ast.literal_eval(x)[0])
primary_cleaned['hasName'] = primary_cleaned['hasName'].apply(lambda x: ast.literal_eval(x)[0])

C:\Users\micah\AppData\Local\Temp\ipykernel_18924\1755678018.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  primary_cleaned['hasSourceFrame'] = primary_cleaned['hasSourceFrame'].apply(lambda x: ast.literal_eval(x)[0])
C:\Users\micah\AppData\Local\Temp\ipykernel_18924\1755678018.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  primary_cleaned['hasTargetFrame'] = primary_cleaned['hasTargetFrame'].apply(lambda x: ast.literal_eval(x)[0])
C:\Users\micah\AppData\Local\Temp\ipykernel_18924\1755678018.py:5:

In [23]:
primary_cleaned

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,hasMappings,hasAlias
12,Metaphor_DISTINGUISHING_IS_DIVIDING,Frame_Dividing,Frame_Distinguishing,"['Example_1700', 'Example_1701', 'Example_1705...",NaN,Primary,NaN,DISTINGUISHING IS DIVIDING,['Metaphor_family_Primary_Metaphors'],NaN,['DISTINGUISHING IS DIVIDING(Sweetser (1987):n...
18,Metaphor_ACTION_IS_MOTION,Frame_Motion,Frame_Action,"['Example_136', 'Example_134', 'Example_132', ...","[""What is the appropriate Source frame here? ...",Primary,NaN,ACTION IS MOTION,"['Metaphor_family_Primary_Metaphors', 'Metapho...","['Mapping_25', 'Mapping_24']","['ACTION IS MOTION(null:null)', 'ACTION IS MOV..."
26,Metaphor_ACQUIRING_IDEAS_IS_EATING,Frame_Eating,Frame_Thinking_about,"['Example_108', 'Example_115', 'Example_107', ...",['See the numerous subcases of this metaphor f...,Primary,['Entailment_10'],ACQUIRING IDEAS IS EATING,"['Metaphor_family_Mind_Metaphors', 'Metaphor_f...","['Mapping_19', 'Mapping_20', 'Mapping_21']",['ACQUIRING IDEAS IS EATING(null:null)']
33,Metaphor_BEING_WEALTHY_IS_BEING_FREE,Frame_Lack_of_impediment_to_motion,Frame_Wealth,"['Example_647', 'Example_649', 'Example_650', ...",NaN,Primary,NaN,BEING WEALTHY IS BEING FREE,"['Metaphor_family_Primary_Metaphors', 'Metapho...",NaN,['BEING WEALTHY IS BEING FREE(null:null)']
38,Metaphor_COMMUNICATING_IS_SHOWING,Frame_Cause_to_see,Frame_Communication,"['Example_1124', 'Example_1117', 'Example_1121...",NaN,Primary,['Entailment_66'],COMMUNICATING IS SHOWING,"['Metaphor_family_Conduit_Metaphor', 'Metaphor...","['Mapping_176', 'Mapping_174', 'Mapping_175', ...",['COMMUNICATING IS SHOWING(null:null)']
...,...,...,...,...,...,...,...,...,...,...,...
844,Metaphor_HAPPINESS_IS_VERTICALITY,Frame_Verticality,Frame_Happiness,"['Example_2821', 'Example_2822', 'Example_2818...",NaN,Primary,"['Entailment_123', 'Entailment_124']",HAPPINESS IS VERTICALITY,"['Metaphor_family_Happiness_Metaphors', 'Metap...",NaN,"['HAPPINESS IS VERTICALITY(Metaphor+:null)', '..."
845,Metaphor_ATTRIBUTES_ARE_POSSESSIONS,Frame_Possession,Frame_Attribute,"['Example_523', 'Example_524', 'Example_521', ...",NaN,Primary,"['Entailment_38', 'Entailment_37', 'Entailment...",ATTRIBUTES ARE POSSESSIONS,['Metaphor_family_Object_Event_Structure_Metap...,"['Mapping_113', 'Mapping_114']",['ATTRIBUTES ARE POSSESSIONS(Grady (1998):null...
861,Metaphor_EXPERIENCE_OF_A_STATE_IS_SPATIAL_RELA...,Frame_Spatial_relation,Frame_Experiencing_a_state,"['Example_2101', 'Example_2098', 'Example_2097...",['This is a non-perspectivized view of the spa...,Primary,"['Entailment_93', 'Entailment_92', 'Entailment...",EXPERIENCE OF A STATE IS SPATIAL RELATION WITH...,"['Metaphor_family_Orientational_Metaphors', 'M...","['Mapping_256', 'Mapping_257', 'Mapping_258']",['EXPERIENCE OF A STATE IS SPATIAL RELATION WI...
864,Metaphor_BECOMING_AWARE_IS_NOTICING,Frame_Noticing,Frame_Becoming_aware,"['Example_555', 'Example_556', 'Example_553', ...",NaN,Primary,NaN,BECOMING AWARE IS NOTICING,"['Metaphor_family_Primary_Metaphors', 'Metapho...",NaN,NaN


In [24]:
primary_cleaned.hasSourceFrame.value_counts()

hasSourceFrame
Frame_Verticality            10
Frame_Object                  7
Frame_Bounded_region          6
Frame_Being_at_a_location     5
Frame_Motion_along_a_path     4
                             ..
Frame_Making                  1
Frame_Smelly                  1
Frame_Fluid_motion            1
Frame_Measuring               1
Frame_Cause_to_perceive       1
Name: count, Length: 82, dtype: int64

In [25]:
primary_cleaned.hasTargetFrame.value_counts()

hasTargetFrame
Frame_Time                      6
Frame_Action                    4
Frame_Causation                 4
Frame_Difficulties_in_action    4
Frame_Thinking_about            3
                               ..
Frame_Change_of_state           1
Frame_Sympathy                  1
Frame_Organization              1
Frame_Abstract_relatedness      1
Frame_Becoming_aware            1
Name: count, Length: 89, dtype: int64

In [26]:
primary_cleaned
import ast

primary_cleaned["hasExample"] = df["hasExample"].apply(
    lambda x: ast.literal_eval(x) if pd.notna(x) else []
)

primary_cleaned

C:\Users\micah\AppData\Local\Temp\ipykernel_18924\636188549.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  primary_cleaned["hasExample"] = df["hasExample"].apply(


,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,hasMappings,hasAlias
12,Metaphor_DISTINGUISHING_IS_DIVIDING,Frame_Dividing,Frame_Distinguishing,"[Example_1700, Example_1701, Example_1705, Exa...",NaN,Primary,NaN,DISTINGUISHING IS DIVIDING,['Metaphor_family_Primary_Metaphors'],NaN,['DISTINGUISHING IS DIVIDING(Sweetser (1987):n...
18,Metaphor_ACTION_IS_MOTION,Frame_Motion,Frame_Action,"[Example_136, Example_134, Example_132, Exampl...","[""What is the appropriate Source frame here? ...",Primary,NaN,ACTION IS MOTION,"['Metaphor_family_Primary_Metaphors', 'Metapho...","['Mapping_25', 'Mapping_24']","['ACTION IS MOTION(null:null)', 'ACTION IS MOV..."
26,Metaphor_ACQUIRING_IDEAS_IS_EATING,Frame_Eating,Frame_Thinking_about,"[Example_108, Example_115, Example_107, Exampl...",['See the numerous subcases of this metaphor f...,Primary,['Entailment_10'],ACQUIRING IDEAS IS EATING,"['Metaphor_family_Mind_Metaphors', 'Metaphor_f...","['Mapping_19', 'Mapping_20', 'Mapping_21']",['ACQUIRING IDEAS IS EATING(null:null)']
33,Metaphor_BEING_WEALTHY_IS_BEING_FREE,Frame_Lack_of_impediment_to_motion,Frame_Wealth,"[Example_647, Example_649, Example_650, Exampl...",NaN,Primary,NaN,BEING WEALTHY IS BEING FREE,"['Metaphor_family_Primary_Metaphors', 'Metapho...",NaN,['BEING WEALTHY IS BEING FREE(null:null)']
38,Metaphor_COMMUNICATING_IS_SHOWING,Frame_Cause_to_see,Frame_Communication,"[Example_1124, Example_1117, Example_1121, Exa...",NaN,Primary,['Entailment_66'],COMMUNICATING IS SHOWING,"['Metaphor_family_Conduit_Metaphor', 'Metaphor...","['Mapping_176', 'Mapping_174', 'Mapping_175', ...",['COMMUNICATING IS SHOWING(null:null)']
...,...,...,...,...,...,...,...,...,...,...,...
844,Metaphor_HAPPINESS_IS_VERTICALITY,Frame_Verticality,Frame_Happiness,"[Example_2821, Example_2822, Example_2818, Exa...",NaN,Primary,"['Entailment_123', 'Entailment_124']",HAPPINESS IS VERTICALITY,"['Metaphor_family_Happiness_Metaphors', 'Metap...",NaN,"['HAPPINESS IS VERTICALITY(Metaphor+:null)', '..."
845,Metaphor_ATTRIBUTES_ARE_POSSESSIONS,Frame_Possession,Frame_Attribute,"[Example_523, Example_524, Example_521, Exampl...",NaN,Primary,"['Entailment_38', 'Entailment_37', 'Entailment...",ATTRIBUTES ARE POSSESSIONS,['Metaphor_family_Object_Event_Structure_Metap...,"['Mapping_113', 'Mapping_114']",['ATTRIBUTES ARE POSSESSIONS(Grady (1998):null...
861,Metaphor_EXPERIENCE_OF_A_STATE_IS_SPATIAL_RELA...,Frame_Spatial_relation,Frame_Experiencing_a_state,"[Example_2101, Example_2098, Example_2097, Exa...",['This is a non-perspectivized view of the spa...,Primary,"['Entailment_93', 'Entailment_92', 'Entailment...",EXPERIENCE OF A STATE IS SPATIAL RELATION WITH...,"['Metaphor_family_Orientational_Metaphors', 'M...","['Mapping_256', 'Mapping_257', 'Mapping_258']",['EXPERIENCE OF A STATE IS SPATIAL RELATION WI...
864,Metaphor_BECOMING_AWARE_IS_NOTICING,Frame_Noticing,Frame_Becoming_aware,"[Example_555, Example_556, Example_553, Exampl...",NaN,Primary,NaN,BECOMING AWARE IS NOTICING,"['Metaphor_family_Primary_Metaphors', 'Metapho...",NaN,NaN


In [27]:
all_examples = primary_cleaned["hasExample"].explode().dropna().unique().tolist()

all_examples

## get all exasmples from primary only so that i can know which constructions are used by primary metaphors

['Example_1700',
 'Example_1701',
 'Example_1705',
 'Example_1704',
 'Example_1702',
 'Example_1703',
 'Example_136',
 'Example_134',
 'Example_132',
 'Example_135',
 'Example_131',
 'Example_133',
 'Example_137',
 'Example_108',
 'Example_115',
 'Example_107',
 'Example_109',
 'Example_111',
 'Example_113',
 'Example_112',
 'Example_110',
 'Example_114',
 'Example_647',
 'Example_649',
 'Example_650',
 'Example_648',
 'Example_646',
 'Example_651',
 'Example_1124',
 'Example_1117',
 'Example_1121',
 'Example_1119',
 'Example_1118',
 'Example_1122',
 'Example_1123',
 'Example_1120',
 'Example_1824',
 'Example_1827',
 'Example_1826',
 'Example_1825',
 'Example_1823',
 'Example_1822',
 'Example_5011',
 'Example_5013',
 'Example_5016',
 'Example_5015',
 'Example_5012',
 'Example_5014',
 'Example_1642',
 'Example_1641',
 'Example_1645',
 'Example_1644',
 'Example_1643',
 'Example_1639',
 'Example_1640',
 'Example_1646',
 'Example_2982',
 'Example_2980',
 'Example_2978',
 'Example_2977',
 '

In [28]:
import json

with open('data/primary_examples.json', 'w') as f:
    json.dump(all_examples, f)

In [29]:
frame_tuples = list(
    zip(
        primary_cleaned["name"].str.replace("Metaphor_", "", regex=False).str.lower(),
        primary_cleaned["hasSourceFrame"].str.replace("Frame_", "", regex=False).str.lower(),
        primary_cleaned["hasTargetFrame"].str.replace("Frame_", "", regex=False).str.lower()
    )
)
frame_tuples

[('distinguishing_is_dividing', 'dividing', 'distinguishing'),
 ('action_is_motion', 'motion', 'action'),
 ('acquiring_ideas_is_eating', 'eating', 'thinking_about'),
 ('being_wealthy_is_being_free', 'lack_of_impediment_to_motion', 'wealth'),
 ('communicating_is_showing', 'cause_to_see', 'communication'),
 ('effects_are_transferred_objects', 'transfer_scenario', 'causation'),
 ('social_group_is_a_bounded_region', 'bounded_region', 'social_group'),
 ('difficulty_is_heaviness', 'weight', 'difficulties_in_action'),
 ('impediments_to_action_are_antagonistic_forces',
  'antagonistic_force',
  'difficulties_in_action'),
 ('words_are_containers', 'containing', 'word'),
 ('opportunities_are_paths', 'physical_path', 'opportunities'),
 ('thinking_is_moving', 'motion', 'thinking_about'),
 ('qualities_are_substances', 'substance', 'quality'),
 ('visual_field_is_a_container', 'containment', 'visual_field'),
 ('arithmetic_is_object_construction', 'object_construction', 'arithmetic'),
 ('important_is_

Checking examples vs constructions

In [30]:
exploded = primary_cleaned.explode("hasExample")

exploded[exploded["hasExample"] == "Example_4661"]

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,hasMappings,hasAlias
757,Metaphor_PSYCHOLOGICAL_HARM_IS_PHYSICAL_INJURY,Frame_Physical_harm,Frame_Psychological_harm,Example_4661,NaN,Primary,NaN,PSYCHOLOGICAL HARM IS PHYSICAL INJURY,"['Metaphor_family_Primary_Metaphors', 'Metapho...",NaN,['PSYCHOLOGICAL HARM IS PHYSICAL INJURY(Oana:n...


In [31]:
exploded['sName'] = exploded['name'].apply(lambda x: x.replace("Metaphor_", "").lower())

exploded

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,hasMappings,hasAlias,sName
12,Metaphor_DISTINGUISHING_IS_DIVIDING,Frame_Dividing,Frame_Distinguishing,Example_1700,NaN,Primary,NaN,DISTINGUISHING IS DIVIDING,['Metaphor_family_Primary_Metaphors'],NaN,['DISTINGUISHING IS DIVIDING(Sweetser (1987):n...,distinguishing_is_dividing
12,Metaphor_DISTINGUISHING_IS_DIVIDING,Frame_Dividing,Frame_Distinguishing,Example_1701,NaN,Primary,NaN,DISTINGUISHING IS DIVIDING,['Metaphor_family_Primary_Metaphors'],NaN,['DISTINGUISHING IS DIVIDING(Sweetser (1987):n...,distinguishing_is_dividing
12,Metaphor_DISTINGUISHING_IS_DIVIDING,Frame_Dividing,Frame_Distinguishing,Example_1705,NaN,Primary,NaN,DISTINGUISHING IS DIVIDING,['Metaphor_family_Primary_Metaphors'],NaN,['DISTINGUISHING IS DIVIDING(Sweetser (1987):n...,distinguishing_is_dividing
12,Metaphor_DISTINGUISHING_IS_DIVIDING,Frame_Dividing,Frame_Distinguishing,Example_1704,NaN,Primary,NaN,DISTINGUISHING IS DIVIDING,['Metaphor_family_Primary_Metaphors'],NaN,['DISTINGUISHING IS DIVIDING(Sweetser (1987):n...,distinguishing_is_dividing
12,Metaphor_DISTINGUISHING_IS_DIVIDING,Frame_Dividing,Frame_Distinguishing,Example_1702,NaN,Primary,NaN,DISTINGUISHING IS DIVIDING,['Metaphor_family_Primary_Metaphors'],NaN,['DISTINGUISHING IS DIVIDING(Sweetser (1987):n...,distinguishing_is_dividing
...,...,...,...,...,...,...,...,...,...,...,...,...
871,Metaphor_COMMUNICATING_IS_POINTING,Frame_Cause_to_perceive,Frame_Communication,Example_1115,NaN,Primary,NaN,COMMUNICATING IS POINTING,"['Metaphor_family_Primary_Metaphors', 'Metapho...",NaN,NaN,communicating_is_pointing
871,Metaphor_COMMUNICATING_IS_POINTING,Frame_Cause_to_perceive,Frame_Communication,Example_1110,NaN,Primary,NaN,COMMUNICATING IS POINTING,"['Metaphor_family_Primary_Metaphors', 'Metapho...",NaN,NaN,communicating_is_pointing
871,Metaphor_COMMUNICATING_IS_POINTING,Frame_Cause_to_perceive,Frame_Communication,Example_1111,NaN,Primary,NaN,COMMUNICATING IS POINTING,"['Metaphor_family_Primary_Metaphors', 'Metapho...",NaN,NaN,communicating_is_pointing
871,Metaphor_COMMUNICATING_IS_POINTING,Frame_Cause_to_perceive,Frame_Communication,Example_1112,NaN,Primary,NaN,COMMUNICATING IS POINTING,"['Metaphor_family_Primary_Metaphors', 'Metapho...",NaN,NaN,communicating_is_pointing


In [32]:
has_both_wordlists = [
    "acquiring_ideas_is_eating",
    "action_is_being_in_a_location",
    "action_is_motion",
    "action_is_motion_along_a_path",
    "action_is_self_propelled_motion_along_a_path",
    "affection_is_warmth",
    "anger_is_heat",
    "argument_is_war",
    "assistance_is_support",
    "bad_is_stinky",
    "being_alive_is_being_physically_at_this_location",
    "beliefs_are_possessions",
    "causes_are_forces",
    "change_in_mood_is_vertical_movement",
    "change_is_acquisition",
    "communicating_is_showing",
    "control_is_a_bounded_region",
    "control_is_verticality",
    "desire_is_hunger",
    "difficulty_is_hardness",
    "difficulty_is_heaviness",
    "difficulty_is_physical_struggle",
    "education_is_a_resource",
    "effects_are_transferred_objects",
    "emotions_are_objects",
    "existing_is_being_physically_at_this_location",
    "functionality_is_verticality",
    "groups_are_containers",
    "happiness_is_verticality",
    "happy_is_light",
    "health_is_verticality",
    "ideas_are_food",
    "ideas_are_objects",
    "impediments_to_action_are_antagonistic_forces",
    "learning_is_acquiring",
    "life_is_verticality",
    "morality_is_a_straight_path",
    "morality_is_straightness",
    "negative_emotional_states_are_physical_pains",
    "now_is_here",
    "ongoing_causation_is_nourishing",
    "opportunities_are_objects",
    "opportunities_are_paths",
    "passion_is_heat",
    "past_is_behind",
    "peaceful_is_motionless",
    "purity_is_cleanliness",
    "quantity_is_size",
    "quantity_is_verticality",
    "relationship_is_a_bounded_region_in_space",
    "relationships_are_objects",
    "resource_is_a_liquid",
    "seeing_is_touching",
    "thinking_is_moving",
    "thinking_is_perceiving",
    "time_is_a_landscape_through_which_the_observer_moves",
    "time_periods_are_containers",
    "well_being_is_verticality"
]

exploded_filtered = exploded[exploded["sName"].isin(has_both_wordlists)]

exploded = exploded_filtered

In [33]:
exploded[exploded["hasExample"] == "Example_979"]

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,hasMappings,hasAlias,sName


In [34]:
exploded[exploded["hasExample"].isin(["Example_159", "Example_2007", "Example_4826"])]

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,hasMappings,hasAlias,sName
606,Metaphor_RESOURCE_IS_A_LIQUID,Frame_Liquid,Frame_Resource,Example_4826,NaN,Primary,NaN,RESOURCE IS A LIQUID,['Metaphor_family_Primary_Metaphors'],NaN,['RESOURCE IS A LIQUID(null:null)'],resource_is_a_liquid


In [35]:
exploded[exploded["hasExample"].isin(["Example_122", "Example_3746", "Example_5486", "Example_4943", "Example_134", "Example_5492", "Example_5489"])]

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,hasMappings,hasAlias,sName
18,Metaphor_ACTION_IS_MOTION,Frame_Motion,Frame_Action,Example_134,"[""What is the appropriate Source frame here? ...",Primary,NaN,ACTION IS MOTION,"['Metaphor_family_Primary_Metaphors', 'Metapho...","['Mapping_25', 'Mapping_24']","['ACTION IS MOTION(null:null)', 'ACTION IS MOV...",action_is_motion
225,Metaphor_SEEING_IS_TOUCHING,Frame_Touching,Frame_Seeing,Example_4943,NaN,Primary,['Entailment_218'],SEEING IS TOUCHING,['Metaphor_family_Grady_Primary_Metaphors'],"['Mapping_538', 'Mapping_536', 'Mapping_535', ...",['VISION IS PHYSICAL CONTACT(Grady (1998):null...,seeing_is_touching
494,Metaphor_ACTION_IS_BEING_IN_A_LOCATION,Frame_Being_at_a_location,Frame_Action,Example_122,['The Target domain for the current metaphor i...,Primary,"['Entailment_14', 'Entailment_12', 'Entailment...",ACTION IS BEING IN A LOCATION,"['Metaphor_family_Primary_Metaphors', 'Metapho...","['Mapping_22', 'Mapping_23']",['PERFORMING AN ACTION IS BEING AT A LOCATION(...,action_is_being_in_a_location


In [36]:
exploded[exploded["hasExample"].isin(["Example_1173", "Example_1172", "Example_1175", "Example_4824", "Example_1633", "Example_167", "Example_1640"])]

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,hasMappings,hasAlias,sName
50,Metaphor_DIFFICULTY_IS_HEAVINESS,Frame_Weight,Frame_Difficulties_in_action,Example_1640,NaN,Primary,NaN,DIFFICULTY IS HEAVINESS,['Metaphor_family_Grady_Primary_Metaphors'],NaN,['DIFFICULTY IS HEAVINESS(Grady (1998):null)'],difficulty_is_heaviness
439,Metaphor_DIFFICULTY_IS_HARDNESS,Frame_Firmness,Frame_Difficulties_in_action,Example_1633,NaN,Primary,NaN,DIFFICULTY IS HARDNESS,['Metaphor_family_Grady_Primary_Metaphors'],NaN,NaN,difficulty_is_hardness
606,Metaphor_RESOURCE_IS_A_LIQUID,Frame_Liquid,Frame_Resource,Example_4824,NaN,Primary,NaN,RESOURCE IS A LIQUID,['Metaphor_family_Primary_Metaphors'],NaN,['RESOURCE IS A LIQUID(null:null)'],resource_is_a_liquid


In [37]:
exploded[exploded['hasExample'].isin(['Example_144',
 'Example_1064',
 'Example_1212',
 'Example_4794',
 'Example_4801',
 'Example_2231',
 'Example_4987',
 'Example_2235',
 'Example_1061',
 'Example_143',
 'Example_3749',
 'Example_1060',
 'Example_2233',
 'Example_1063',
 'Example_3748',
 'Example_2234',
 'Example_4988',
 'Example_2236',
 'Example_4986',
 'Example_2232'])]

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,hasMappings,hasAlias,sName
263,Metaphor_FUNCTIONALITY_IS_VERTICALITY,Frame_Verticality,Frame_Functionality,Example_2232,['We need to decide if we want to distinguish ...,Primary,"['Entailment_104', 'Entailment_106', 'Entailme...",FUNCTIONALITY IS VERTICALITY,"['Metaphor_family_Grady_Primary_Metaphors', 'M...","['Mapping_278', 'Mapping_276', 'Mapping_277']",['FUNCTIONALITY IS ERECTNESS(Grady (1998):null...,functionality_is_verticality
263,Metaphor_FUNCTIONALITY_IS_VERTICALITY,Frame_Verticality,Frame_Functionality,Example_2231,['We need to decide if we want to distinguish ...,Primary,"['Entailment_104', 'Entailment_106', 'Entailme...",FUNCTIONALITY IS VERTICALITY,"['Metaphor_family_Grady_Primary_Metaphors', 'M...","['Mapping_278', 'Mapping_276', 'Mapping_277']",['FUNCTIONALITY IS ERECTNESS(Grady (1998):null...,functionality_is_verticality
263,Metaphor_FUNCTIONALITY_IS_VERTICALITY,Frame_Verticality,Frame_Functionality,Example_2235,['We need to decide if we want to distinguish ...,Primary,"['Entailment_104', 'Entailment_106', 'Entailme...",FUNCTIONALITY IS VERTICALITY,"['Metaphor_family_Grady_Primary_Metaphors', 'M...","['Mapping_278', 'Mapping_276', 'Mapping_277']",['FUNCTIONALITY IS ERECTNESS(Grady (1998):null...,functionality_is_verticality
263,Metaphor_FUNCTIONALITY_IS_VERTICALITY,Frame_Verticality,Frame_Functionality,Example_2236,['We need to decide if we want to distinguish ...,Primary,"['Entailment_104', 'Entailment_106', 'Entailme...",FUNCTIONALITY IS VERTICALITY,"['Metaphor_family_Grady_Primary_Metaphors', 'M...","['Mapping_278', 'Mapping_276', 'Mapping_277']",['FUNCTIONALITY IS ERECTNESS(Grady (1998):null...,functionality_is_verticality
263,Metaphor_FUNCTIONALITY_IS_VERTICALITY,Frame_Verticality,Frame_Functionality,Example_2233,['We need to decide if we want to distinguish ...,Primary,"['Entailment_104', 'Entailment_106', 'Entailme...",FUNCTIONALITY IS VERTICALITY,"['Metaphor_family_Grady_Primary_Metaphors', 'M...","['Mapping_278', 'Mapping_276', 'Mapping_277']",['FUNCTIONALITY IS ERECTNESS(Grady (1998):null...,functionality_is_verticality
263,Metaphor_FUNCTIONALITY_IS_VERTICALITY,Frame_Verticality,Frame_Functionality,Example_2234,['We need to decide if we want to distinguish ...,Primary,"['Entailment_104', 'Entailment_106', 'Entailme...",FUNCTIONALITY IS VERTICALITY,"['Metaphor_family_Grady_Primary_Metaphors', 'M...","['Mapping_278', 'Mapping_276', 'Mapping_277']",['FUNCTIONALITY IS ERECTNESS(Grady (1998):null...,functionality_is_verticality
451,Metaphor_ACTION_IS_MOTION_ALONG_A_PATH,Frame_Motion_along_a_path,Frame_Action,Example_143,['What used to be ACTION IS MOTION or ACTION I...,Primary,"['Entailment_17', 'Entailment_16', 'Entailment...",ACTION IS MOTION ALONG A PATH,['Metaphor_family_Location_Event_Structure_Met...,"['Mapping_29', 'Mapping_26', 'Mapping_27', 'Ma...","['ACTION IS MOTION(null:null)', 'ACTION IS MOV...",action_is_motion_along_a_path
451,Metaphor_ACTION_IS_MOTION_ALONG_A_PATH,Frame_Motion_along_a_path,Frame_Action,Example_144,['What used to be ACTION IS MOTION or ACTION I...,Primary,"['Entailment_17', 'Entailment_16', 'Entailment...",ACTION IS MOTION ALONG A PATH,['Metaphor_family_Location_Event_Structure_Met...,"['Mapping_29', 'Mapping_26', 'Mapping_27', 'Ma...","['ACTION IS MOTION(null:null)', 'ACTION IS MOV...",action_is_motion_along_a_path
667,Metaphor_RELATIONSHIPS_ARE_OBJECTS,Frame_Object,Frame_Relationship,Example_4801,['This is a supraordinate-level metaphor for a...,Primary,NaN,RELATIONSHIPS ARE OBJECTS,"['Metaphor_family_LOVE_IS_A_JOURNEY_Cascade', ...",['Mapping_505'],['RELATIONSHIPS ARE OBJECTS(Metaphor+:null)'],relationships_are_objects
667,Metaphor_RELATIONSHIPS_ARE_OBJECTS,Frame_Object,Frame_Relationship,Example_4794,['This is a supraordinate-level metaphor for a...,Primary,NaN,RELATIONSHIPS ARE OBJECTS,"['Metaphor_family_LOVE_IS_A_JOURNEY_Cascade', ...",['Mapping_505'],['RELAT

In [38]:
exploded[exploded['hasExample'].isin(['Example_3395',
 'Example_495',
 'Example_5141',
 'Example_5465',
 'Example_2982',
 'Example_2981',
 'Example_5015',
 'Example_151',
 'Example_506',
 'Example_1066',
 'Example_4160',
 'Example_2898',
 'Example_5468',
 'Example_4163',
 'Example_2895',
 'Example_4659',
 'Example_2980',
 'Example_1065',
 'Example_1642',
 'Example_1643',
 'Example_502',
 'Example_4159'])]

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,hasMappings,hasAlias,sName
50,Metaphor_DIFFICULTY_IS_HEAVINESS,Frame_Weight,Frame_Difficulties_in_action,Example_1642,NaN,Primary,NaN,DIFFICULTY IS HEAVINESS,['Metaphor_family_Grady_Primary_Metaphors'],NaN,['DIFFICULTY IS HEAVINESS(Grady (1998):null)'],difficulty_is_heaviness
50,Metaphor_DIFFICULTY_IS_HEAVINESS,Frame_Weight,Frame_Difficulties_in_action,Example_1643,NaN,Primary,NaN,DIFFICULTY IS HEAVINESS,['Metaphor_family_Grady_Primary_Metaphors'],NaN,['DIFFICULTY IS HEAVINESS(Grady (1998):null)'],difficulty_is_heaviness
72,Metaphor_IMPEDIMENTS_TO_ACTION_ARE_ANTAGONISTI...,Frame_Antagonistic_force,Frame_Difficulties_in_action,Example_2982,NaN,Primary,NaN,IMPEDIMENTS TO ACTION ARE ANTAGONISTIC FORCES,['Metaphor_family_Location_Event_Structure_Met...,NaN,['IMPEDIMENTS TO ACTION ARE ANTAGONISTIC FORCE...,impediments_to_action_are_antagonistic_forces
72,Metaphor_IMPEDIMENTS_TO_ACTION_ARE_ANTAGONISTI...,Frame_Antagonistic_force,Frame_Difficulties_in_action,Example_2980,NaN,Primary,NaN,IMPEDIMENTS TO ACTION ARE ANTAGONISTIC FORCES,['Metaphor_family_Location_Event_Structure_Met...,NaN,['IMPEDIMENTS TO ACTION ARE ANTAGONISTIC FORCE...,impediments_to_action_are_antagonistic_forces
72,Metaphor_IMPEDIMENTS_TO_ACTION_ARE_ANTAGONISTI...,Frame_Antagonistic_force,Frame_Difficulties_in_action,Example_2981,NaN,Primary,NaN,IMPEDIMENTS TO ACTION ARE ANTAGONISTIC FORCES,['Metaphor_family_Location_Event_Structure_Met...,NaN,['IMPEDIMENTS TO ACTION ARE ANTAGONISTIC FORCE...,impediments_to_action_are_antagonistic_forces
83,Metaphor_OPPORTUNITIES_ARE_PATHS,Frame_Physical_path,Frame_Opportunities,Example_4159,['see discussion'],Primary,NaN,OPPORTUNITIES ARE PATHS,['Metaphor_family_Location_Event_Structure_Met...,NaN,['OPPORTUNITIES ARE PATHS(Oana:null)'],opportunities_are_paths
83,Metaphor_OPPORTUNITIES_ARE_PATHS,Frame_Physical_path,Frame_Opportunities,Example_4160,['see discussion'],Primary,NaN,OPPORTUNITIES ARE PATHS,['Metaphor_family_Location_Event_Structure_Met...,NaN,['OPPORTUNITIES ARE PATHS(Oana:null)'],opportunities_are_paths
83,Metaphor_OPPORTUNITIES_ARE_PATHS,Frame_Physical_path,Frame_Opportunities,Example_4163,['see discussion'],Primary,NaN,OPPORTUNITIES ARE PATHS,['Metaphor_family_Location_Event_Structure_Met...,NaN,['OPPORTUNITIES ARE PATHS(Oana:null)'],opportunities_are_paths
195,Metaphor_ASSISTANCE_IS_SUPPORT,Frame_Support,Frame_Assistance,Example_495,NaN,Primary,NaN,ASSISTANCE IS SUPPORT,['Metaphor_family_Grady_Primary_Metaphors'],NaN,"['HELP IS SUPPORT(null:null)', 'ASSISTANCE IS ...",assistance_is_support
284,Metaphor_IDEAS_ARE_FOOD,Frame_Food,Frame_Idea,Example_2898,['Merged with KNOWLEDGE IS FOOD'],Primary,"['Entailment_128', 'Entailment_133', 'Entailme...",IDEAS ARE FOOD,"['Metaphor_family_Primary_Metaphors', 'Metapho...","['Mapping_350', 'Mapping_349']","['KNOWLEDGE IS FOOD(null:null)', 'IDEAS ARE FO...",ideas_are_food


In [39]:
exploded[exploded["hasExample"].isin(['Example_1163',
 'Example_471',
 'Example_4657',
 'Example_985',
 'Example_1790',
 'Example_5145',
 'Example_1789',
 'Example_4098',
 'Example_4231',
 'Example_2000',
 'Example_4653',
 'Example_1788',
 'Example_1915',
 'Example_4664',
 'Example_5187',
 'Example_1787',
 'Example_5188',
 'Example_4230',
 'Example_1162',
 'Example_4651',
 'Example_988',
 'Example_4095',
 'Example_467',
 'Example_4655',
 'Example_5140',
 'Example_4654',
 'Example_5186'])]

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,hasMappings,hasAlias,sName
476,Metaphor_PEACEFUL_IS_MOTIONLESS,Frame_Motionless,Frame_Peacefulness,Example_4230,NaN,Primary,NaN,PEACEFUL IS MOTIONLESS,['Metaphor_family_Emotion_Metaphors'],NaN,['PEACEFUL IS MOTIONLESS(null:null)'],peaceful_is_motionless
476,Metaphor_PEACEFUL_IS_MOTIONLESS,Frame_Motionless,Frame_Peacefulness,Example_4231,NaN,Primary,NaN,PEACEFUL IS MOTIONLESS,['Metaphor_family_Emotion_Metaphors'],NaN,['PEACEFUL IS MOTIONLESS(null:null)'],peaceful_is_motionless
558,Metaphor_ARGUMENT_IS_WAR,Frame_War,Frame_Argument,Example_471,NaN,Primary,NaN,ARGUMENT IS WAR,['Metaphor_family_Primary_Metaphors'],NaN,['ARGUMENT IS WAR(Metaphors We Live By:null)'],argument_is_war
558,Metaphor_ARGUMENT_IS_WAR,Frame_War,Frame_Argument,Example_467,NaN,Primary,NaN,ARGUMENT IS WAR,['Metaphor_family_Primary_Metaphors'],NaN,['ARGUMENT IS WAR(Metaphors We Live By:null)'],argument_is_war
565,Metaphor_NEGATIVE_EMOTIONAL_STATES_ARE_PHYSICA...,Frame_Experience_pain,Frame_Emotion,Example_4098,NaN,Primary,NaN,NEGATIVE EMOTIONAL STATES ARE PHYSICAL PAINS,['Metaphor_family_Emotion_Metaphors'],['Mapping_453'],NaN,negative_emotional_states_are_physical_pains
565,Metaphor_NEGATIVE_EMOTIONAL_STATES_ARE_PHYSICA...,Frame_Experience_pain,Frame_Emotion,Example_4095,NaN,Primary,NaN,NEGATIVE EMOTIONAL STATES ARE PHYSICAL PAINS,['Metaphor_family_Emotion_Metaphors'],['Mapping_453'],NaN,negative_emotional_states_are_physical_pains
746,Metaphor_EDUCATION_IS_A_RESOURCE,Frame_Resource,Frame_Education,Example_1787,NaN,Primary,NaN,EDUCATION IS A RESOURCE,['Metaphor_family_Education_metaphors'],"['Mapping_222', 'Mapping_223']",NaN,education_is_a_resource
746,Metaphor_EDUCATION_IS_A_RESOURCE,Frame_Resource,Frame_Education,Example_1789,NaN,Primary,NaN,EDUCATION IS A RESOURCE,['Metaphor_family_Education_metaphors'],"['Mapping_222', 'Mapping_223']",NaN,education_is_a_resource
746,Metaphor_EDUCATION_IS_A_RESOURCE,Frame_Resource,Frame_Education,Example_1790,NaN,Primary,NaN,EDUCATION IS A RESOURCE,['Metaphor_family_Education_metaphors'],"['Mapping_222', 'Mapping_223']",NaN,education_is_a_resource
746,Metaphor_EDUCATION_IS_A_RESOURCE,Frame_Resource,Frame_Education,Example_1788,NaN,Primary,NaN,EDUCATION IS A RESOURCE,['Metaphor_family_Education_metaphors'],"['Mapping_222', 'Mapping_223']",NaN,education_is_a_resource


In [40]:
exploded[exploded["hasExample"].isin(['Example_4099',
 'Example_493',
 'Example_5491',
 'Example_5380',
 'Example_4194',
 'Example_4217',
 'Example_5377',
 'Example_4388',
 'Example_978',
 'Example_4719',
 'Example_2688',
 'Example_466',
 'Example_4383',
 'Example_5488',
 'Example_2687',
 'Example_4715',
 'Example_468',
 'Example_441',
 'Example_3908',
 'Example_5128',
 'Example_4198',
 'Example_2125',
 'Example_4387',
 'Example_469',
 'Example_2691',
 'Example_548',
 'Example_5122',
 'Example_2831',
 'Example_4386',
 'Example_4191',
 'Example_437',
 'Example_4713',
 'Example_488',
 'Example_2692',
 'Example_4384',
 'Example_4385',
 'Example_490',
 'Example_2690',
 'Example_2689',
 'Example_3249',
 'Example_438',
 'Example_4218',
 'Example_2123',
 'Example_5378',
 'Example_5119',
 'Example_491',
 'Example_470',
 'Example_3906',
 'Example_5375',
 'Example_5381',
 'Example_4097'])]

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,hasMappings,hasAlias,sName
143,Metaphor_HAPPY_IS_LIGHT,Frame_Lightness,Frame_Happiness,Example_2831,NaN,Primary,NaN,HAPPY IS LIGHT,"['Metaphor_family_Emotion_Metaphors', 'Metapho...",NaN,NaN,happy_is_light
327,Metaphor_BAD_IS_STINKY,Frame_Smelly,Frame_Badness,Example_548,['Not necessarily the right source domain (Dis...,Primary,NaN,BAD IS STINKY,['Metaphor_family_Grady_Primary_Metaphors'],NaN,['BAD IS FOUL-SMELLING(Grady (1998):null)'],bad_is_stinky
547,Metaphor_QUANTITY_IS_SIZE,Frame_Size,Frame_Quantity,Example_4715,NaN,Primary,"['Entailment_207', 'Entailment_206']",QUANTITY IS SIZE,"['Metaphor_family_Grady_Primary_Metaphors', 'M...","['Mapping_492', 'Mapping_495', 'Mapping_493', ...","['AMOUNT IS SIZE(null:null)', 'QUANTITY IS VOL...",quantity_is_size
547,Metaphor_QUANTITY_IS_SIZE,Frame_Size,Frame_Quantity,Example_4713,NaN,Primary,"['Entailment_207', 'Entailment_206']",QUANTITY IS SIZE,"['Metaphor_family_Grady_Primary_Metaphors', 'M...","['Mapping_492', 'Mapping_495', 'Mapping_493', ...","['AMOUNT IS SIZE(null:null)', 'QUANTITY IS VOL...",quantity_is_size
547,Metaphor_QUANTITY_IS_SIZE,Frame_Size,Frame_Quantity,Example_4719,NaN,Primary,"['Entailment_207', 'Entailment_206']",QUANTITY IS SIZE,"['Metaphor_family_Grady_Primary_Metaphors', 'M...","['Mapping_492', 'Mapping_495', 'Mapping_493', ...","['AMOUNT IS SIZE(null:null)', 'QUANTITY IS VOL...",quantity_is_size
558,Metaphor_ARGUMENT_IS_WAR,Frame_War,Frame_Argument,Example_469,NaN,Primary,NaN,ARGUMENT IS WAR,['Metaphor_family_Primary_Metaphors'],NaN,['ARGUMENT IS WAR(Metaphors We Live By:null)'],argument_is_war
558,Metaphor_ARGUMENT_IS_WAR,Frame_War,Frame_Argument,Example_470,NaN,Primary,NaN,ARGUMENT IS WAR,['Metaphor_family_Primary_Metaphors'],NaN,['ARGUMENT IS WAR(Metaphors We Live By:null)'],argument_is_war
558,Metaphor_ARGUMENT_IS_WAR,Frame_War,Frame_Argument,Example_466,NaN,Primary,NaN,ARGUMENT IS WAR,['Metaphor_family_Primary_Metaphors'],NaN,['ARGUMENT IS WAR(Metaphors We Live By:null)'],argument_is_war
558,Metaphor_ARGUMENT_IS_WAR,Frame_War,Frame_Argument,Example_468,NaN,Primary,NaN,ARGUMENT IS WAR,['Metaphor_family_Primary_Metaphors'],NaN,['ARGUMENT IS WAR(Metaphors We Live By:null)'],argument_is_war
565,Metaphor_NEGATIVE_EMOTIONAL_STATES_ARE_PHYSICA...,Frame_Experience_pain,Frame_Emotion,Example_4099,NaN,Primary,NaN,NEGATIVE EMOTIONAL STATES ARE PHYSICAL PAINS,['Metaphor_family_Emotion_Metaphors'],['Mapping_453'],NaN,negative_emotional_states_are_physical_pains


In [41]:
exploded[exploded["hasExample"].isin(['Example_504',
 'Example_153',
 'Example_1213',
 'Example_155',
 'Example_1112',
 'Example_1280',
 'Example_3393',
 'Example_498',
 'Example_5484',
 'Example_2901',
 'Example_125',
 'Example_5353',
 'Example_3910',
 'Example_1236',
 'Example_154',
 'Example_500',
 'Example_463',
 'Example_1069',
 'Example_1111',
 'Example_3907',
 'Example_146',
 'Example_128',
 'Example_503',
 'Example_5480',
 'Example_157',
 'Example_1088',
 'Example_5478',
 'Example_126',
 'Example_2977',
 'Example_3905',
 'Example_1238',
 'Example_5324',
 'Example_732',
 'Example_5603',
 'Example_5014',
 'Example_2833',
 'Example_152',
 'Example_5327',
 'Example_1234',
 'Example_1927',
 'Example_3578',
 'Example_147',
 'Example_5597',
 'Example_658',
 'Example_652',
 'Example_1649',
 'Example_1211',
 'Example_1285',
 'Example_3386',
 'Example_137',
 'Example_4142',
 'Example_2861',
 'Example_169',
 'Example_5012',
 'Example_5483',
 'Example_5313',
 'Example_5139',
 'Example_3574',
 'Example_5546',
 'Example_5355',
 'Example_148',
 'Example_150',
 'Example_3909',
 'Example_160',
 'Example_1650',
 'Example_1242',
 'Example_653',
 'Example_4153',
 'Example_2863',
 'Example_1206',
 'Example_141',
 'Example_4144',
 'Example_981',
 'Example_149',
 'Example_4930',
 'Example_5479',
 'Example_1207',
 'Example_133',
 'Example_435',
 'Example_1084',
 'Example_156',
 'Example_5490',
 'Example_2862'])]

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,hasMappings,hasAlias,sName
18,Metaphor_ACTION_IS_MOTION,Frame_Motion,Frame_Action,Example_133,"[""What is the appropriate Source frame here? ...",Primary,NaN,ACTION IS MOTION,"['Metaphor_family_Primary_Metaphors', 'Metapho...","['Mapping_25', 'Mapping_24']","['ACTION IS MOTION(null:null)', 'ACTION IS MOV...",action_is_motion
18,Metaphor_ACTION_IS_MOTION,Frame_Motion,Frame_Action,Example_137,"[""What is the appropriate Source frame here? ...",Primary,NaN,ACTION IS MOTION,"['Metaphor_family_Primary_Metaphors', 'Metapho...","['Mapping_25', 'Mapping_24']","['ACTION IS MOTION(null:null)', 'ACTION IS MOV...",action_is_motion
72,Metaphor_IMPEDIMENTS_TO_ACTION_ARE_ANTAGONISTI...,Frame_Antagonistic_force,Frame_Difficulties_in_action,Example_2977,NaN,Primary,NaN,IMPEDIMENTS TO ACTION ARE ANTAGONISTIC FORCES,['Metaphor_family_Location_Event_Structure_Met...,NaN,['IMPEDIMENTS TO ACTION ARE ANTAGONISTIC FORCE...,impediments_to_action_are_antagonistic_forces
88,Metaphor_THINKING_IS_MOVING,Frame_Motion,Frame_Thinking_about,Example_5313,NaN,Primary,"['Entailment_247', 'Entailment_246', 'Entailme...",THINKING IS MOVING,['Metaphor_family_Mind_Metaphors'],NaN,['THOUGHT IS MOTION(null:null)'],thinking_is_moving
143,Metaphor_HAPPY_IS_LIGHT,Frame_Lightness,Frame_Happiness,Example_2833,NaN,Primary,NaN,HAPPY IS LIGHT,"['Metaphor_family_Emotion_Metaphors', 'Metapho...",NaN,NaN,happy_is_light
195,Metaphor_ASSISTANCE_IS_SUPPORT,Frame_Support,Frame_Assistance,Example_498,NaN,Primary,NaN,ASSISTANCE IS SUPPORT,['Metaphor_family_Grady_Primary_Metaphors'],NaN,"['HELP IS SUPPORT(null:null)', 'ASSISTANCE IS ...",assistance_is_support
195,Metaphor_ASSISTANCE_IS_SUPPORT,Frame_Support,Frame_Assistance,Example_500,NaN,Primary,NaN,ASSISTANCE IS SUPPORT,['Metaphor_family_Grady_Primary_Metaphors'],NaN,"['HELP IS SUPPORT(null:null)', 'ASSISTANCE IS ...",assistance_is_support
248,Metaphor_EMOTIONS_ARE_OBJECTS,Frame_Object,Frame_Emotion,Example_1927,['This is a supraordinate source frame to cove...,Primary,NaN,EMOTIONS ARE OBJECTS,"['Metaphor_family_Primary_Metaphors', 'Metapho...",['Mapping_241'],['EMOTIONS ARE OBJECTS(Jisup:null)'],emotions_are_objects
266,Metaphor_WELL_BEING_IS_VERTICALITY,Frame_Verticality,Frame_Well_being,Example_5546,NaN,Primary,"['Entailment_256', 'Entailment_257']",WELL-BEING IS VERTICALITY,"['Metaphor_family_Orientational_Metaphors', 'M...",NaN,"['WELL-BEING IS VERTICALITY(null:null)', 'WELL...",well_being_is_verticality
322,Metaphor_ONGOING_CAUSATION_IS_NOURISHING,Frame_Food,Frame_Causation,Example_4144,NaN,Primary,NaN,ONGOING CAUSATION IS NOURISHING,NaN,NaN,['ONGOING CAUSATION IS NOURISHING(null:null)'],ongoing_causation_is_nourishing


In [42]:
exploded[exploded["hasExample"].isin(['Example_4767',
 'Example_2905',
 'Example_4720',
 'Example_4764',
 'Example_5404',
 'Example_4989',
 'Example_4798',
 'Example_2870',
 'Example_4154',
 'Example_4096',
 'Example_4227',
 'Example_4658',
 'Example_5142',
 'Example_5184',
 'Example_5602',
 'Example_4228',
 'Example_2852',
 'Example_240',
 'Example_245',
 'Example_4682',
 'Example_413',
 'Example_5600',
 'Example_1644',
 'Example_1911',
 'Example_4685',
 'Example_5604',
 'Example_5405',
 'Example_1636',
 'Example_416',
 'Example_489',
 'Example_5601',
 'Example_1916',
 'Example_492',
 'Example_4718',
 'Example_2008',
 'Example_5138',
 'Example_2893',
 'Example_4796',
 'Example_4665',
 'Example_1647',
 'Example_3812',
 'Example_4795',
 'Example_646',
 'Example_4714',
 'Example_5400',
 'Example_5403',
 'Example_5147',
 'Example_4717',
 'Example_1639',
 'Example_2850',
 'Example_4797',
 'Example_4684',
 'Example_4768',
 'Example_420',
 'Example_647',
 'Example_550',
 'Example_4100',
 'Example_5185',
 'Example_1235',
 'Example_5143',
 'Example_2849',
 'Example_417',
 'Example_4681',
 'Example_547',
 'Example_5402',
 'Example_1912',
 'Example_4726',
 'Example_4155',
 'Example_415',
 'Example_2848',
 'Example_4215',
 'Example_3079',
 'Example_241',
 'Example_3808',
 'Example_4728',
 'Example_5452',
 'Example_4214',
 'Example_4216',
 'Example_1652',
 'Example_418',
 'Example_4769',
 'Example_2851',
 'Example_4765',
 'Example_1641',
 'Example_4716',
 'Example_1632',
 'Example_4990',
 'Example_244',
 'Example_4799',
 'Example_4985',
 'Example_243'])]

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,hasMappings,hasAlias,sName
50,Metaphor_DIFFICULTY_IS_HEAVINESS,Frame_Weight,Frame_Difficulties_in_action,Example_1641,NaN,Primary,NaN,DIFFICULTY IS HEAVINESS,['Metaphor_family_Grady_Primary_Metaphors'],NaN,['DIFFICULTY IS HEAVINESS(Grady (1998):null)'],difficulty_is_heaviness
50,Metaphor_DIFFICULTY_IS_HEAVINESS,Frame_Weight,Frame_Difficulties_in_action,Example_1644,NaN,Primary,NaN,DIFFICULTY IS HEAVINESS,['Metaphor_family_Grady_Primary_Metaphors'],NaN,['DIFFICULTY IS HEAVINESS(Grady (1998):null)'],difficulty_is_heaviness
50,Metaphor_DIFFICULTY_IS_HEAVINESS,Frame_Weight,Frame_Difficulties_in_action,Example_1639,NaN,Primary,NaN,DIFFICULTY IS HEAVINESS,['Metaphor_family_Grady_Primary_Metaphors'],NaN,['DIFFICULTY IS HEAVINESS(Grady (1998):null)'],difficulty_is_heaviness
170,Metaphor_AFFECTION_IS_WARMTH,Frame_Warmth,Frame_Affectionate_action,Example_240,NaN,Primary,NaN,AFFECTION IS WARMTH,"['Metaphor_family_Primary_Metaphors', 'Metapho...",NaN,['AFFECTION IS WARMTH(null:null)'],affection_is_warmth
170,Metaphor_AFFECTION_IS_WARMTH,Frame_Warmth,Frame_Affectionate_action,Example_243,NaN,Primary,NaN,AFFECTION IS WARMTH,"['Metaphor_family_Primary_Metaphors', 'Metapho...",NaN,['AFFECTION IS WARMTH(null:null)'],affection_is_warmth
170,Metaphor_AFFECTION_IS_WARMTH,Frame_Warmth,Frame_Affectionate_action,Example_245,NaN,Primary,NaN,AFFECTION IS WARMTH,"['Metaphor_family_Primary_Metaphors', 'Metapho...",NaN,['AFFECTION IS WARMTH(null:null)'],affection_is_warmth
170,Metaphor_AFFECTION_IS_WARMTH,Frame_Warmth,Frame_Affectionate_action,Example_244,NaN,Primary,NaN,AFFECTION IS WARMTH,"['Metaphor_family_Primary_Metaphors', 'Metapho...",NaN,['AFFECTION IS WARMTH(null:null)'],affection_is_warmth
170,Metaphor_AFFECTION_IS_WARMTH,Frame_Warmth,Frame_Affectionate_action,Example_241,NaN,Primary,NaN,AFFECTION IS WARMTH,"['Metaphor_family_Primary_Metaphors', 'Metapho...",NaN,['AFFECTION IS WARMTH(null:null)'],affection_is_warmth
243,Metaphor_TIME_PERIODS_ARE_CONTAINERS,Frame_Containment,Frame_Time,Example_5400,['Resolve source and target frames.'],Primary,NaN,TIME PERIODS ARE CONTAINERS,"['Metaphor_family_Time_Metaphors', 'Metaphor_f...",NaN,['TIME PERIODS IN OUR LIVES ARE CONTAINERS(Gra...,time_periods_are_containers
243,Metaphor_TIME_PERIODS_ARE_CONTAINERS,Frame_Containment,Frame_Time,Example_5405,['Resolve source and target frames.'],Primary,NaN,TIME PERIODS ARE CONTAINERS,"['Metaphor_family_Time_Metaphors', 'Metaphor_f...",NaN,['TIME PERIODS IN OUR LIVES ARE CONTAINERS(Gra...,time_periods_are_containers


In [43]:
exploded[exploded["hasExample"].isin(['Example_4724',
 'Example_2821',
 'Example_648',
 'Example_5570',
 'Example_5360',
 'Example_5364',
 'Example_5363',
 'Example_5562',
 'Example_5362',
 'Example_4725',
 'Example_2001',
 'Example_545',
 'Example_865',
 'Example_4977',
 'Example_816',
 'Example_5358',
 'Example_4825',
 'Example_815',
 'Example_871',
 'Example_5573',
 'Example_4197',
 'Example_3320',
 'Example_4942',
 'Example_544',
 'Example_5567',
 'Example_5566',
 'Example_145',
 'Example_727',
 'Example_939',
 'Example_1087',
 'Example_2040',
 'Example_1119',
 'Example_5357',
 'Example_869',
 'Example_2820',
 'Example_4929',
 'Example_4193',
 'Example_4926',
 'Example_5365',
 'Example_2039',
 'Example_4924',
 'Example_2036',
 'Example_4937',
 'Example_940',
 'Example_1164',
 'Example_4660',
 'Example_1161',
 'Example_4490',
 'Example_4925',
 'Example_5564',
 'Example_2042',
 'Example_2683',
 'Example_2904',
 'Example_5366',
 'Example_730',
 'Example_5575',
 'Example_2035',
 'Example_2978',
 'Example_4656',
 'Example_813',
 'Example_5379',
 'Example_566',
 'Example_4939',
 'Example_5572',
 'Example_161',
 'Example_866',
 'Example_1160',
 'Example_1906',
 'Example_1120',
 'Example_5563',
 'Example_4941',
 'Example_2682',
 'Example_5361',
 'Example_4491',
 'Example_817',
 'Example_872',
 'Example_5565',
 'Example_5382',
 'Example_3322',
 'Example_4488',
 'Example_5574',
 'Example_4823',
 'Example_5359',
 'Example_1113',
 'Example_4938',
 'Example_4722',
 'Example_2685',
 'Example_2903',
 'Example_2684',
 'Example_5571',
 'Example_3321',
 'Example_4927',
 'Example_4928',
 'Example_2037',
 'Example_2038',
 'Example_814',
 'Example_496',
 'Example_3323',
 'Example_864',
 'Example_4923',
 'Example_2681'])]

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,hasMappings,hasAlias,sName
38,Metaphor_COMMUNICATING_IS_SHOWING,Frame_Cause_to_see,Frame_Communication,Example_1119,NaN,Primary,['Entailment_66'],COMMUNICATING IS SHOWING,"['Metaphor_family_Conduit_Metaphor', 'Metaphor...","['Mapping_176', 'Mapping_174', 'Mapping_175', ...",['COMMUNICATING IS SHOWING(null:null)'],communicating_is_showing
38,Metaphor_COMMUNICATING_IS_SHOWING,Frame_Cause_to_see,Frame_Communication,Example_1120,NaN,Primary,['Entailment_66'],COMMUNICATING IS SHOWING,"['Metaphor_family_Conduit_Metaphor', 'Metaphor...","['Mapping_176', 'Mapping_174', 'Mapping_175', ...",['COMMUNICATING IS SHOWING(null:null)'],communicating_is_showing
72,Metaphor_IMPEDIMENTS_TO_ACTION_ARE_ANTAGONISTI...,Frame_Antagonistic_force,Frame_Difficulties_in_action,Example_2978,NaN,Primary,NaN,IMPEDIMENTS TO ACTION ARE ANTAGONISTIC FORCES,['Metaphor_family_Location_Event_Structure_Met...,NaN,['IMPEDIMENTS TO ACTION ARE ANTAGONISTIC FORCE...,impediments_to_action_are_antagonistic_forces
176,Metaphor_CAUSES_ARE_FORCES,Frame_Force,Frame_Causation,Example_871,['see Discussion'],Primary,NaN,CAUSES ARE FORCES,['Metaphor_family_Location_Event_Structure_Met...,NaN,['CAUSES ARE FORCES(PITF:null)'],causes_are_forces
176,Metaphor_CAUSES_ARE_FORCES,Frame_Force,Frame_Causation,Example_866,['see Discussion'],Primary,NaN,CAUSES ARE FORCES,['Metaphor_family_Location_Event_Structure_Met...,NaN,['CAUSES ARE FORCES(PITF:null)'],causes_are_forces
176,Metaphor_CAUSES_ARE_FORCES,Frame_Force,Frame_Causation,Example_865,['see Discussion'],Primary,NaN,CAUSES ARE FORCES,['Metaphor_family_Location_Event_Structure_Met...,NaN,['CAUSES ARE FORCES(PITF:null)'],causes_are_forces
176,Metaphor_CAUSES_ARE_FORCES,Frame_Force,Frame_Causation,Example_872,['see Discussion'],Primary,NaN,CAUSES ARE FORCES,['Metaphor_family_Location_Event_Structure_Met...,NaN,['CAUSES ARE FORCES(PITF:null)'],causes_are_forces
176,Metaphor_CAUSES_ARE_FORCES,Frame_Force,Frame_Causation,Example_864,['see Discussion'],Primary,NaN,CAUSES ARE FORCES,['Metaphor_family_Location_Event_Structure_Met...,NaN,['CAUSES ARE FORCES(PITF:null)'],causes_are_forces
176,Metaphor_CAUSES_ARE_FORCES,Frame_Force,Frame_Causation,Example_869,['see Discussion'],Primary,NaN,CAUSES ARE FORCES,['Metaphor_family_Location_Event_Structure_Met...,NaN,['CAUSES ARE FORCES(PITF:null)'],causes_are_forces
195,Metaphor_ASSISTANCE_IS_SUPPORT,Frame_Support,Frame_Assistance,Example_496,NaN,Primary,NaN,ASSISTANCE IS SUPPORT,['Metaphor_family_Grady_Primary_Metaphors'],NaN,"['HELP IS SUPPORT(null:null)', 'ASSISTANCE IS ...",assistance_is_support


In [44]:
exploded[exploded["hasExample"].isin(['Example_549',
 'Example_1824',
 'Example_3490',
 'Example_3390',
 'Example_4711',
 'Example_436',
 'Example_1827',
 'Example_937',
 'Example_4650',
 'Example_2006',
 'Example_948',
 'Example_482',
 'Example_3492',
 'Example_3389',
 'Example_5309',
 'Example_5469',
 'Example_4707',
 'Example_518',
 'Example_4123',
 'Example_108',
 'Example_2128',
 'Example_3247',
 'Example_5126',
 'Example_2873',
 'Example_163',
 'Example_946',
 'Example_4139',
 'Example_557',
 'Example_138',
 'Example_941',
 'Example_1785',
 'Example_3489',
 'Example_654',
 'Example_5466',
 'Example_523',
 'Example_475',
 'Example_1826',
 'Example_1786',
 'Example_5125',
 'Example_525',
 'Example_552',
 'Example_3493',
 'Example_2874',
 'Example_485',
 'Example_3246',
 'Example_553',
 'Example_4141',
 'Example_555',
 'Example_520',
 'Example_4157',
 'Example_3488',
 'Example_472',
 'Example_2836',
 'Example_1922',
 'Example_2126',
 'Example_5304',
 'Example_657',
 'Example_1929',
 'Example_481',
 'Example_4487',
 'Example_4489',
 'Example_4827',
 'Example_728',
 'Example_1924',
 'Example_4121',
 'Example_656',
 'Example_942',
 'Example_1822',
 'Example_2187',
 'Example_3750',
 'Example_4229',
 'Example_5326',
 'Example_4710',
 'Example_140',
 'Example_5306',
 'Example_521',
 'Example_1823',
 'Example_3243',
 'Example_5303',
 'Example_945',
 'Example_114',
 'Example_4143',
 'Example_556',
 'Example_4158',
 'Example_4125',
 'Example_807',
 'Example_943',
 'Example_526',
 'Example_655',
 'Example_1068',
 'Example_1072',
 'Example_3394',
 'Example_5311',
 'Example_486',
 'Example_4122',
 'Example_944',
 'Example_949',
 'Example_2872',
 'Example_2002',
 'Example_2832',
 'Example_947',
 'Example_1928',
 'Example_4140',
 'Example_1070',
 'Example_3392',
 'Example_938',
 'Example_5449',
 'Example_1085',
 'Example_551',
 'Example_5305',
 'Example_5464',
 'Example_111',
 'Example_5127',
 'Example_474',
 'Example_5329',
 'Example_4828',
 'Example_2847',
 'Example_1923',
 'Example_484',
 'Example_4712',
 'Example_4940',
 'Example_731',
 'Example_136',
 'Example_440',
 'Example_1073',
 'Example_3494',
 'Example_2041',
 'Example_4492',
 'Example_1117',
 'Example_5307',
 'Example_5334',
 'Example_5016',
 'Example_162',
 'Example_2186',
 'Example_4120',
 'Example_1086',
 'Example_1825',
 'Example_139',
 'Example_2130',
 'Example_478',
 'Example_3245',
 'Example_1089',
 'Example_476',
 'Example_519',
 'Example_135',
 'Example_158',
 'Example_818',
 'Example_5314',
 'Example_2127',
 'Example_2005',
 'Example_2131',
 'Example_4156',
 'Example_5338',
 'Example_4124',
 'Example_3491',
 'Example_726',
 'Example_5450',
 'Example_2818',
 'Example_5455',
 'Example_554',
 'Example_3242',
 'Example_5310',
 'Example_524',
 'Example_5451',
 'Example_477',
 'Example_565',
 'Example_5467',
 'Example_5470',
 'Example_473',
 'Example_522',
 'Example_569',
 'Example_2902',
 'Example_5401',
 'Example_5325',
 'Example_5312',
 'Example_2900',
 'Example_110',
 'Example_4800',
 'Example_5129',
 'Example_142'])]

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,hasMappings,hasAlias,sName
18,Metaphor_ACTION_IS_MOTION,Frame_Motion,Frame_Action,Example_136,"[""What is the appropriate Source frame here? ...",Primary,NaN,ACTION IS MOTION,"['Metaphor_family_Primary_Metaphors', 'Metapho...","['Mapping_25', 'Mapping_24']","['ACTION IS MOTION(null:null)', 'ACTION IS MOV...",action_is_motion
18,Metaphor_ACTION_IS_MOTION,Frame_Motion,Frame_Action,Example_135,"[""What is the appropriate Source frame here? ...",Primary,NaN,ACTION IS MOTION,"['Metaphor_family_Primary_Metaphors', 'Metapho...","['Mapping_25', 'Mapping_24']","['ACTION IS MOTION(null:null)', 'ACTION IS MOV...",action_is_motion
26,Metaphor_ACQUIRING_IDEAS_IS_EATING,Frame_Eating,Frame_Thinking_about,Example_108,['See the numerous subcases of this metaphor f...,Primary,['Entailment_10'],ACQUIRING IDEAS IS EATING,"['Metaphor_family_Mind_Metaphors', 'Metaphor_f...","['Mapping_19', 'Mapping_20', 'Mapping_21']",['ACQUIRING IDEAS IS EATING(null:null)'],acquiring_ideas_is_eating
26,Metaphor_ACQUIRING_IDEAS_IS_EATING,Frame_Eating,Frame_Thinking_about,Example_111,['See the numerous subcases of this metaphor f...,Primary,['Entailment_10'],ACQUIRING IDEAS IS EATING,"['Metaphor_family_Mind_Metaphors', 'Metaphor_f...","['Mapping_19', 'Mapping_20', 'Mapping_21']",['ACQUIRING IDEAS IS EATING(null:null)'],acquiring_ideas_is_eating
26,Metaphor_ACQUIRING_IDEAS_IS_EATING,Frame_Eating,Frame_Thinking_about,Example_110,['See the numerous subcases of this metaphor f...,Primary,['Entailment_10'],ACQUIRING IDEAS IS EATING,"['Metaphor_family_Mind_Metaphors', 'Metaphor_f...","['Mapping_19', 'Mapping_20', 'Mapping_21']",['ACQUIRING IDEAS IS EATING(null:null)'],acquiring_ideas_is_eating
...,...,...,...,...,...,...,...,...,...,...,...,...
791,Metaphor_LEARNING_IS_ACQUIRING,Frame_Gain_possession,Frame_Learning,Example_3492,NaN,Primary,NaN,LEARNING IS ACQUIRING,"['Metaphor_family_Conduit_Metaphor', 'Metaphor...","['Mapping_398', 'Mapping_396', 'Mapping_397']",['LEARNING IS ACQUIRING(Grady (1998):127)'],learning_is_acquiring
791,Metaphor_LEARNING_IS_ACQUIRING,Frame_Gain_possession,Frame_Learning,Example_3489,NaN,Primary,NaN,LEARNING IS ACQUIRING,"['Metaphor_family_Conduit_Metaphor', 'Metaphor...","['Mapping_398', 'Mapping_396', 'Mapping_397']",['LEARNING IS ACQUIRING(Grady (1998):127)'],learning_is_acquiring
791,Metaphor_LEARNING_IS_ACQUIRING,Frame_Gain_possession,Frame_Learning,Example_3493,NaN,Primary,NaN,LEARNING IS ACQUIRING,"['Metaphor_family_Conduit_Metaphor', 'Metaphor...","['Mapping_398', 'Mapping_396', 'Mapping_397']",['LEARNING IS ACQUIRING(Grady (1998):127)'],learning_is_acquiring
791,Metaphor_LEARNING_IS_ACQUIRING,Frame_Gain_possession,Frame_Learning,Example_3491,NaN,Primary,NaN,LEARNING IS ACQUIRING,"['Metaphor_family_Conduit_Metaphor', 'Metaphor...","['Mapping_398', 'Mapping_396', 'Mapping_397']",['LEARNING IS ACQUIRING(Grady (1998):127)'],learning_is_acquiring


In [46]:
exploded

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,hasMappings,hasAlias,sName
18,Metaphor_ACTION_IS_MOTION,Frame_Motion,Frame_Action,Example_136,"[""What is the appropriate Source frame here? ...",Primary,NaN,ACTION IS MOTION,"['Metaphor_family_Primary_Metaphors', 'Metapho...","['Mapping_25', 'Mapping_24']","['ACTION IS MOTION(null:null)', 'ACTION IS MOV...",action_is_motion
18,Metaphor_ACTION_IS_MOTION,Frame_Motion,Frame_Action,Example_134,"[""What is the appropriate Source frame here? ...",Primary,NaN,ACTION IS MOTION,"['Metaphor_family_Primary_Metaphors', 'Metapho...","['Mapping_25', 'Mapping_24']","['ACTION IS MOTION(null:null)', 'ACTION IS MOV...",action_is_motion
18,Metaphor_ACTION_IS_MOTION,Frame_Motion,Frame_Action,Example_132,"[""What is the appropriate Source frame here? ...",Primary,NaN,ACTION IS MOTION,"['Metaphor_family_Primary_Metaphors', 'Metapho...","['Mapping_25', 'Mapping_24']","['ACTION IS MOTION(null:null)', 'ACTION IS MOV...",action_is_motion
18,Metaphor_ACTION_IS_MOTION,Frame_Motion,Frame_Action,Example_135,"[""What is the appropriate Source frame here? ...",Primary,NaN,ACTION IS MOTION,"['Metaphor_family_Primary_Metaphors', 'Metapho...","['Mapping_25', 'Mapping_24']","['ACTION IS MOTION(null:null)', 'ACTION IS MOV...",action_is_motion
18,Metaphor_ACTION_IS_MOTION,Frame_Motion,Frame_Action,Example_131,"[""What is the appropriate Source frame here? ...",Primary,NaN,ACTION IS MOTION,"['Metaphor_family_Primary_Metaphors', 'Metapho...","['Mapping_25', 'Mapping_24']","['ACTION IS MOTION(null:null)', 'ACTION IS MOV...",action_is_motion
...,...,...,...,...,...,...,...,...,...,...,...,...
844,Metaphor_HAPPINESS_IS_VERTICALITY,Frame_Verticality,Frame_Happiness,Example_2822,NaN,Primary,"['Entailment_123', 'Entailment_124']",HAPPINESS IS VERTICALITY,"['Metaphor_family_Happiness_Metaphors', 'Metap...",NaN,"['HAPPINESS IS VERTICALITY(Metaphor+:null)', '...",happiness_is_verticality
844,Metaphor_HAPPINESS_IS_VERTICALITY,Frame_Verticality,Frame_Happiness,Example_2818,NaN,Primary,"['Entailment_123', 'Entailment_124']",HAPPINESS IS VERTICALITY,"['Metaphor_family_Happiness_Metaphors', 'Metap...",NaN,"['HAPPINESS IS VERTICALITY(Metaphor+:null)', '...",happiness_is_verticality
844,Metaphor_HAPPINESS_IS_VERTICALITY,Frame_Verticality,Frame_Happiness,Example_2819,NaN,Primary,"['Entailment_123', 'Entailment_124']",HAPPINESS IS VERTICALITY,"['Metaphor_family_Happiness_Metaphors', 'Metap...",NaN,"['HAPPINESS IS VERTICALITY(Metaphor+:null)', '...",happiness_is_verticality
844,Metaphor_HAPPINESS_IS_VERTICALITY,Frame_Verticality,Frame_Happiness,Example_2820,NaN,Primary,"['Entailment_123', 'Entailment_124']",HAPPINESS IS VERTICALITY,"['Metaphor_family_Happiness_Metaphors', 'Metap...",NaN,"['HAPPINESS IS VERTICALITY(Metaphor+:null)', '...",happiness_is_verticality
